From 038490c6ce3f72fa76873e49c5f985de963f2681 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Mon, 14 May 2018 14:12:18 +0800
Subject: [PATCH 022/129] media: intel-ipu4: imx319: Fix coding style issues

Message for Open Source:
Fix coding style issues.

Message for Internal:
[Issue/Feature] Coding style issue.
[Root Cause/Changes] Add spaces in single line braces.

Change-Id: I35f4c1225b91d12e5fead742f5a20916ca0717f2
Tracked-On: #JICCCN-171
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/i2c/imx319.c | 2953 ++++++++++++++++++++++----------------------
 1 file changed, 1475 insertions(+), 1478 deletions(-)

diff --git a/drivers/media/i2c/imx319.c b/drivers/media/i2c/imx319.c
index 9f57c4f..75a717b 100644
--- a/drivers/media/i2c/imx319.c
+++ b/drivers/media/i2c/imx319.c
@@ -91,110 +91,110 @@ struct imx319_mode {
 };
 
 static const struct imx319_reg imx319_global_regs[] = {
-	{0x0136, 0x13},
-	{0x0137, 0x33},
-	{0x3c7e, 0x03},
-	{0x3c7f, 0x07},
-	{0x4d39, 0x0b},
-	{0x4d41, 0x33},
-	{0x4d43, 0x0c},
-	{0x4d49, 0x89},
-	{0x4e05, 0x0b},
-	{0x4e0d, 0x33},
-	{0x4e0f, 0x0c},
-	{0x4e15, 0x89},
-	{0x4e49, 0x2a},
-	{0x4e51, 0x33},
-	{0x4e53, 0x0c},
-	{0x4e59, 0x89},
-	{0x5601, 0x4f},
-	{0x560b, 0x45},
-	{0x562f, 0x0a},
-	{0x5643, 0x0a},
-	{0x5645, 0x0c},
-	{0x56ef, 0x51},
-	{0x586f, 0x33},
-	{0x5873, 0x89},
-	{0x5905, 0x33},
-	{0x5907, 0x89},
-	{0x590d, 0x33},
-	{0x590f, 0x89},
-	{0x5915, 0x33},
-	{0x5917, 0x89},
-	{0x5969, 0x1c},
-	{0x596b, 0x72},
-	{0x5971, 0x33},
-	{0x5973, 0x89},
-	{0x5975, 0x33},
-	{0x5977, 0x89},
-	{0x5979, 0x1c},
-	{0x597b, 0x72},
-	{0x5985, 0x33},
-	{0x5987, 0x89},
-	{0x5999, 0x1c},
-	{0x599b, 0x72},
-	{0x59a5, 0x33},
-	{0x59a7, 0x89},
-	{0x7485, 0x08},
-	{0x7487, 0x0c},
-	{0x7489, 0xc7},
-	{0x748b, 0x8b},
-	{0x9004, 0x09},
-	{0x9200, 0x6a},
-	{0x9201, 0x22},
-	{0x9202, 0x6a},
-	{0x9203, 0x23},
-	{0x9204, 0x5f},
-	{0x9205, 0x23},
-	{0x9206, 0x5f},
-	{0x9207, 0x24},
-	{0x9208, 0x5f},
-	{0x9209, 0x26},
-	{0x920a, 0x5f},
-	{0x920b, 0x27},
-	{0x920c, 0x5f},
-	{0x920d, 0x29},
-	{0x920e, 0x5f},
-	{0x920f, 0x2a},
-	{0x9210, 0x5f},
-	{0x9211, 0x2c},
-	{0xbc22, 0x1a},
-	{0xf01f, 0x04},
-	{0xf021, 0x03},
-	{0xf023, 0x02},
-	{0xf03d, 0x05},
-	{0xf03f, 0x03},
-	{0xf041, 0x02},
-	{0xf0af, 0x04},
-	{0xf0b1, 0x03},
-	{0xf0b3, 0x02},
-	{0xf0cd, 0x05},
-	{0xf0cf, 0x03},
-	{0xf0d1, 0x02},
-	{0xf13f, 0x04},
-	{0xf141, 0x03},
-	{0xf143, 0x02},
-	{0xf15d, 0x05},
-	{0xf15f, 0x03},
-	{0xf161, 0x02},
-	{0xf1cf, 0x04},
-	{0xf1d1, 0x03},
-	{0xf1d3, 0x02},
-	{0xf1ed, 0x05},
-	{0xf1ef, 0x03},
-	{0xf1f1, 0x02},
-	{0xf287, 0x04},
-	{0xf289, 0x03},
-	{0xf28b, 0x02},
-	{0xf2a5, 0x05},
-	{0xf2a7, 0x03},
-	{0xf2a9, 0x02},
-	{0xf2b7, 0x04},
-	{0xf2b9, 0x03},
-	{0xf2bb, 0x02},
-	{0xf2d5, 0x05},
-	{0xf2d7, 0x03},
-	{0xf2d9, 0x02},
+	{ 0x0136, 0x13 },
+	{ 0x0137, 0x33 },
+	{ 0x3c7e, 0x03 },
+	{ 0x3c7f, 0x07 },
+	{ 0x4d39, 0x0b },
+	{ 0x4d41, 0x33 },
+	{ 0x4d43, 0x0c },
+	{ 0x4d49, 0x89 },
+	{ 0x4e05, 0x0b },
+	{ 0x4e0d, 0x33 },
+	{ 0x4e0f, 0x0c },
+	{ 0x4e15, 0x89 },
+	{ 0x4e49, 0x2a },
+	{ 0x4e51, 0x33 },
+	{ 0x4e53, 0x0c },
+	{ 0x4e59, 0x89 },
+	{ 0x5601, 0x4f },
+	{ 0x560b, 0x45 },
+	{ 0x562f, 0x0a },
+	{ 0x5643, 0x0a },
+	{ 0x5645, 0x0c },
+	{ 0x56ef, 0x51 },
+	{ 0x586f, 0x33 },
+	{ 0x5873, 0x89 },
+	{ 0x5905, 0x33 },
+	{ 0x5907, 0x89 },
+	{ 0x590d, 0x33 },
+	{ 0x590f, 0x89 },
+	{ 0x5915, 0x33 },
+	{ 0x5917, 0x89 },
+	{ 0x5969, 0x1c },
+	{ 0x596b, 0x72 },
+	{ 0x5971, 0x33 },
+	{ 0x5973, 0x89 },
+	{ 0x5975, 0x33 },
+	{ 0x5977, 0x89 },
+	{ 0x5979, 0x1c },
+	{ 0x597b, 0x72 },
+	{ 0x5985, 0x33 },
+	{ 0x5987, 0x89 },
+	{ 0x5999, 0x1c },
+	{ 0x599b, 0x72 },
+	{ 0x59a5, 0x33 },
+	{ 0x59a7, 0x89 },
+	{ 0x7485, 0x08 },
+	{ 0x7487, 0x0c },
+	{ 0x7489, 0xc7 },
+	{ 0x748b, 0x8b },
+	{ 0x9004, 0x09 },
+	{ 0x9200, 0x6a },
+	{ 0x9201, 0x22 },
+	{ 0x9202, 0x6a },
+	{ 0x9203, 0x23 },
+	{ 0x9204, 0x5f },
+	{ 0x9205, 0x23 },
+	{ 0x9206, 0x5f },
+	{ 0x9207, 0x24 },
+	{ 0x9208, 0x5f },
+	{ 0x9209, 0x26 },
+	{ 0x920a, 0x5f },
+	{ 0x920b, 0x27 },
+	{ 0x920c, 0x5f },
+	{ 0x920d, 0x29 },
+	{ 0x920e, 0x5f },
+	{ 0x920f, 0x2a },
+	{ 0x9210, 0x5f },
+	{ 0x9211, 0x2c },
+	{ 0xbc22, 0x1a },
+	{ 0xf01f, 0x04 },
+	{ 0xf021, 0x03 },
+	{ 0xf023, 0x02 },
+	{ 0xf03d, 0x05 },
+	{ 0xf03f, 0x03 },
+	{ 0xf041, 0x02 },
+	{ 0xf0af, 0x04 },
+	{ 0xf0b1, 0x03 },
+	{ 0xf0b3, 0x02 },
+	{ 0xf0cd, 0x05 },
+	{ 0xf0cf, 0x03 },
+	{ 0xf0d1, 0x02 },
+	{ 0xf13f, 0x04 },
+	{ 0xf141, 0x03 },
+	{ 0xf143, 0x02 },
+	{ 0xf15d, 0x05 },
+	{ 0xf15f, 0x03 },
+	{ 0xf161, 0x02 },
+	{ 0xf1cf, 0x04 },
+	{ 0xf1d1, 0x03 },
+	{ 0xf1d3, 0x02 },
+	{ 0xf1ed, 0x05 },
+	{ 0xf1ef, 0x03 },
+	{ 0xf1f1, 0x02 },
+	{ 0xf287, 0x04 },
+	{ 0xf289, 0x03 },
+	{ 0xf28b, 0x02 },
+	{ 0xf2a5, 0x05 },
+	{ 0xf2a7, 0x03 },
+	{ 0xf2a9, 0x02 },
+	{ 0xf2b7, 0x04 },
+	{ 0xf2b9, 0x03 },
+	{ 0xf2bb, 0x02 },
+	{ 0xf2d5, 0x05 },
+	{ 0xf2d7, 0x03 },
+	{ 0xf2d9, 0x02 },
 };
 
 struct imx319_reg_list imx319_global_setting = {
@@ -203,1395 +203,1395 @@ struct imx319_reg_list imx319_global_setting = {
 };
 
 static const struct imx319_reg mode_3264x2448_regs[] = {
-	{0x0112, 0x0a},
-	{0x0113, 0x0a},
-	{0x0114, 0x03},
-	{0x0342, 0x0f},
-	{0x0343, 0x80},
-	{0x0340, 0x0a},
-	{0x0341, 0x78},
-	{0x0344, 0x00},
-	{0x0345, 0x00},
-	{0x0346, 0x00},
-	{0x0347, 0x00},
-	{0x0348, 0x0c},
-	{0x0349, 0xcf},
-	{0x034a, 0x09},
-	{0x034b, 0x9f},
-	{0x0220, 0x00},
-	{0x0221, 0x11},
-	{0x0381, 0x01},
-	{0x0383, 0x01},
-	{0x0385, 0x01},
-	{0x0387, 0x01},
-	{0x0900, 0x00},
-	{0x0901, 0x11},
-	{0x0902, 0x0a},
-	{0x3140, 0x02},
-	{0x3141, 0x00},
-	{0x3f0d, 0x0a},
-	{0x3f14, 0x01},
-	{0x3f3c, 0x01},
-	{0x3f4d, 0x01},
-	{0x3f4c, 0x01},
-	{0x4254, 0x7f},
-	{0x0401, 0x00},
-	{0x0404, 0x00},
-	{0x0405, 0x10},
-	{0x0408, 0x00},
-	{0x0409, 0x08},
-	{0x040a, 0x00},
-	{0x040b, 0x08},
-	{0x040c, 0x0c},
-	{0x040d, 0xc0},
-	{0x040e, 0x09},
-	{0x040f, 0x90},
-	{0x034c, 0x0c},
-	{0x034d, 0xc0},
-	{0x034e, 0x09},
-	{0x034f, 0x90},
-	{0x3261, 0x00},
-	{0x3264, 0x00},
-	{0x3265, 0x10},
-	{0x0301, 0x06},
-	{0x0303, 0x04},
-	{0x0305, 0x03},
-	{0x0306, 0x01},
-	{0x0307, 0x2c},
-	{0x0309, 0x0a},
-	{0x030b, 0x02},
-	{0x030d, 0x04},
-	{0x030e, 0x01},
-	{0x030f, 0x2c},
-	{0x0310, 0x01},
-	{0x0820, 0x0b},
-	{0x0821, 0x40},
-	{0x0822, 0x00},
-	{0x0823, 0x00},
-	{0x3e20, 0x01},
-	{0x3e37, 0x00},
-	{0x3e3b, 0x01},
-	{0x38a3, 0x01},
-	{0x38a8, 0x00},
-	{0x38a9, 0x00},
-	{0x38aa, 0x00},
-	{0x38ab, 0x00},
-	{0x3234, 0x00},
-	{0x3fc1, 0x00},
-	{0x3235, 0x00},
-	{0x3802, 0x00},
-	{0x3143, 0x04},
-	{0x360a, 0x00},
-	{0x0b00, 0x00},
-	{0x0106, 0x00},
-	{0x0b05, 0x01},
-	{0x0b06, 0x01},
-	{0x3230, 0x00},
-	{0x3602, 0x01},
-	{0x3607, 0x01},
-	{0x3c00, 0x00},
-	{0x3c01, 0x48},
-	{0x3c02, 0xc8},
-	{0x3c03, 0xaa},
-	{0x3c04, 0x91},
-	{0x3c05, 0x54},
-	{0x3c06, 0x26},
-	{0x3c07, 0x20},
-	{0x3c08, 0x51},
-	{0x3d80, 0x00},
-	{0x3f50, 0x00},
-	{0x3f56, 0x00},
-	{0x3f57, 0x30},
-	{0x3f78, 0x01},
-	{0x3f79, 0x18},
-	{0x3f7c, 0x00},
-	{0x3f7d, 0x00},
-	{0x3fba, 0x00},
-	{0x3fbb, 0x00},
-	{0xa081, 0x00},
-	{0xe014, 0x00},
-	{0x0202, 0x0a},
-	{0x0203, 0x66},
-	{0x0224, 0x01},
-	{0x0225, 0xf4},
-	{0x0204, 0x00},
-	{0x0205, 0x00},
-	{0x0216, 0x00},
-	{0x0217, 0x00},
-	{0x020e, 0x01},
-	{0x020f, 0x00},
-	{0x0210, 0x01},
-	{0x0211, 0x00},
-	{0x0212, 0x01},
-	{0x0213, 0x00},
-	{0x0214, 0x01},
-	{0x0215, 0x00},
-	{0x0218, 0x01},
-	{0x0219, 0x00},
-	{0x3614, 0x00},
-	{0x3616, 0x0d},
-	{0x3617, 0x56},
-	{0xb612, 0x20},
-	{0xb613, 0x20},
-	{0xb614, 0x20},
-	{0xb615, 0x20},
-	{0xb616, 0x0a},
-	{0xb617, 0x0a},
-	{0xb618, 0x20},
-	{0xb619, 0x20},
-	{0xb61a, 0x20},
-	{0xb61b, 0x20},
-	{0xb61c, 0x0a},
-	{0xb61d, 0x0a},
-	{0xb666, 0x30},
-	{0xb667, 0x30},
-	{0xb668, 0x30},
-	{0xb669, 0x30},
-	{0xb66a, 0x14},
-	{0xb66b, 0x14},
-	{0xb66c, 0x20},
-	{0xb66d, 0x20},
-	{0xb66e, 0x20},
-	{0xb66f, 0x20},
-	{0xb670, 0x10},
-	{0xb671, 0x10},
-	{0x3237, 0x00},
-	{0x3900, 0x00},
-	{0x3901, 0x00},
-	{0x3902, 0x00},
-	{0x3904, 0x00},
-	{0x3905, 0x00},
-	{0x3906, 0x00},
-	{0x3907, 0x00},
-	{0x3908, 0x00},
-	{0x3909, 0x00},
-	{0x3912, 0x00},
-	{0x3930, 0x00},
-	{0x3931, 0x00},
-	{0x3933, 0x00},
-	{0x3934, 0x00},
-	{0x3935, 0x00},
-	{0x3936, 0x00},
-	{0x3937, 0x00},
-	{0x30ac, 0x00},
+	{ 0x0112, 0x0a },
+	{ 0x0113, 0x0a },
+	{ 0x0114, 0x03 },
+	{ 0x0342, 0x0f },
+	{ 0x0343, 0x80 },
+	{ 0x0340, 0x0a },
+	{ 0x0341, 0x78 },
+	{ 0x0344, 0x00 },
+	{ 0x0345, 0x00 },
+	{ 0x0346, 0x00 },
+	{ 0x0347, 0x00 },
+	{ 0x0348, 0x0c },
+	{ 0x0349, 0xcf },
+	{ 0x034a, 0x09 },
+	{ 0x034b, 0x9f },
+	{ 0x0220, 0x00 },
+	{ 0x0221, 0x11 },
+	{ 0x0381, 0x01 },
+	{ 0x0383, 0x01 },
+	{ 0x0385, 0x01 },
+	{ 0x0387, 0x01 },
+	{ 0x0900, 0x00 },
+	{ 0x0901, 0x11 },
+	{ 0x0902, 0x0a },
+	{ 0x3140, 0x02 },
+	{ 0x3141, 0x00 },
+	{ 0x3f0d, 0x0a },
+	{ 0x3f14, 0x01 },
+	{ 0x3f3c, 0x01 },
+	{ 0x3f4d, 0x01 },
+	{ 0x3f4c, 0x01 },
+	{ 0x4254, 0x7f },
+	{ 0x0401, 0x00 },
+	{ 0x0404, 0x00 },
+	{ 0x0405, 0x10 },
+	{ 0x0408, 0x00 },
+	{ 0x0409, 0x08 },
+	{ 0x040a, 0x00 },
+	{ 0x040b, 0x08 },
+	{ 0x040c, 0x0c },
+	{ 0x040d, 0xc0 },
+	{ 0x040e, 0x09 },
+	{ 0x040f, 0x90 },
+	{ 0x034c, 0x0c },
+	{ 0x034d, 0xc0 },
+	{ 0x034e, 0x09 },
+	{ 0x034f, 0x90 },
+	{ 0x3261, 0x00 },
+	{ 0x3264, 0x00 },
+	{ 0x3265, 0x10 },
+	{ 0x0301, 0x06 },
+	{ 0x0303, 0x04 },
+	{ 0x0305, 0x03 },
+	{ 0x0306, 0x01 },
+	{ 0x0307, 0x2c },
+	{ 0x0309, 0x0a },
+	{ 0x030b, 0x02 },
+	{ 0x030d, 0x04 },
+	{ 0x030e, 0x01 },
+	{ 0x030f, 0x2c },
+	{ 0x0310, 0x01 },
+	{ 0x0820, 0x0b },
+	{ 0x0821, 0x40 },
+	{ 0x0822, 0x00 },
+	{ 0x0823, 0x00 },
+	{ 0x3e20, 0x01 },
+	{ 0x3e37, 0x00 },
+	{ 0x3e3b, 0x01 },
+	{ 0x38a3, 0x01 },
+	{ 0x38a8, 0x00 },
+	{ 0x38a9, 0x00 },
+	{ 0x38aa, 0x00 },
+	{ 0x38ab, 0x00 },
+	{ 0x3234, 0x00 },
+	{ 0x3fc1, 0x00 },
+	{ 0x3235, 0x00 },
+	{ 0x3802, 0x00 },
+	{ 0x3143, 0x04 },
+	{ 0x360a, 0x00 },
+	{ 0x0b00, 0x00 },
+	{ 0x0106, 0x00 },
+	{ 0x0b05, 0x01 },
+	{ 0x0b06, 0x01 },
+	{ 0x3230, 0x00 },
+	{ 0x3602, 0x01 },
+	{ 0x3607, 0x01 },
+	{ 0x3c00, 0x00 },
+	{ 0x3c01, 0x48 },
+	{ 0x3c02, 0xc8 },
+	{ 0x3c03, 0xaa },
+	{ 0x3c04, 0x91 },
+	{ 0x3c05, 0x54 },
+	{ 0x3c06, 0x26 },
+	{ 0x3c07, 0x20 },
+	{ 0x3c08, 0x51 },
+	{ 0x3d80, 0x00 },
+	{ 0x3f50, 0x00 },
+	{ 0x3f56, 0x00 },
+	{ 0x3f57, 0x30 },
+	{ 0x3f78, 0x01 },
+	{ 0x3f79, 0x18 },
+	{ 0x3f7c, 0x00 },
+	{ 0x3f7d, 0x00 },
+	{ 0x3fba, 0x00 },
+	{ 0x3fbb, 0x00 },
+	{ 0xa081, 0x00 },
+	{ 0xe014, 0x00 },
+	{ 0x0202, 0x0a },
+	{ 0x0203, 0x66 },
+	{ 0x0224, 0x01 },
+	{ 0x0225, 0xf4 },
+	{ 0x0204, 0x00 },
+	{ 0x0205, 0x00 },
+	{ 0x0216, 0x00 },
+	{ 0x0217, 0x00 },
+	{ 0x020e, 0x01 },
+	{ 0x020f, 0x00 },
+	{ 0x0210, 0x01 },
+	{ 0x0211, 0x00 },
+	{ 0x0212, 0x01 },
+	{ 0x0213, 0x00 },
+	{ 0x0214, 0x01 },
+	{ 0x0215, 0x00 },
+	{ 0x0218, 0x01 },
+	{ 0x0219, 0x00 },
+	{ 0x3614, 0x00 },
+	{ 0x3616, 0x0d },
+	{ 0x3617, 0x56 },
+	{ 0xb612, 0x20 },
+	{ 0xb613, 0x20 },
+	{ 0xb614, 0x20 },
+	{ 0xb615, 0x20 },
+	{ 0xb616, 0x0a },
+	{ 0xb617, 0x0a },
+	{ 0xb618, 0x20 },
+	{ 0xb619, 0x20 },
+	{ 0xb61a, 0x20 },
+	{ 0xb61b, 0x20 },
+	{ 0xb61c, 0x0a },
+	{ 0xb61d, 0x0a },
+	{ 0xb666, 0x30 },
+	{ 0xb667, 0x30 },
+	{ 0xb668, 0x30 },
+	{ 0xb669, 0x30 },
+	{ 0xb66a, 0x14 },
+	{ 0xb66b, 0x14 },
+	{ 0xb66c, 0x20 },
+	{ 0xb66d, 0x20 },
+	{ 0xb66e, 0x20 },
+	{ 0xb66f, 0x20 },
+	{ 0xb670, 0x10 },
+	{ 0xb671, 0x10 },
+	{ 0x3237, 0x00 },
+	{ 0x3900, 0x00 },
+	{ 0x3901, 0x00 },
+	{ 0x3902, 0x00 },
+	{ 0x3904, 0x00 },
+	{ 0x3905, 0x00 },
+	{ 0x3906, 0x00 },
+	{ 0x3907, 0x00 },
+	{ 0x3908, 0x00 },
+	{ 0x3909, 0x00 },
+	{ 0x3912, 0x00 },
+	{ 0x3930, 0x00 },
+	{ 0x3931, 0x00 },
+	{ 0x3933, 0x00 },
+	{ 0x3934, 0x00 },
+	{ 0x3935, 0x00 },
+	{ 0x3936, 0x00 },
+	{ 0x3937, 0x00 },
+	{ 0x30ac, 0x00 },
 };
 
 static const struct imx319_reg mode_3280x2464_regs[] = {
-	{0x0112, 0x0a},
-	{0x0113, 0x0a},
-	{0x0114, 0x03},
-	{0x0342, 0x0f},
-	{0x0343, 0x80},
-	{0x0340, 0x0a},
-	{0x0341, 0x78},
-	{0x0344, 0x00},
-	{0x0345, 0x00},
-	{0x0346, 0x00},
-	{0x0347, 0x00},
-	{0x0348, 0x0c},
-	{0x0349, 0xcf},
-	{0x034a, 0x09},
-	{0x034b, 0x9f},
-	{0x0220, 0x00},
-	{0x0221, 0x11},
-	{0x0381, 0x01},
-	{0x0383, 0x01},
-	{0x0385, 0x01},
-	{0x0387, 0x01},
-	{0x0900, 0x00},
-	{0x0901, 0x11},
-	{0x0902, 0x0a},
-	{0x3140, 0x02},
-	{0x3141, 0x00},
-	{0x3f0d, 0x0a},
-	{0x3f14, 0x01},
-	{0x3f3c, 0x01},
-	{0x3f4d, 0x01},
-	{0x3f4c, 0x01},
-	{0x4254, 0x7f},
-	{0x0401, 0x00},
-	{0x0404, 0x00},
-	{0x0405, 0x10},
-	{0x0408, 0x00},
-	{0x0409, 0x00},
-	{0x040a, 0x00},
-	{0x040b, 0x00},
-	{0x040c, 0x0c},
-	{0x040d, 0xd0},
-	{0x040e, 0x09},
-	{0x040f, 0xa0},
-	{0x034c, 0x0c},
-	{0x034d, 0xd0},
-	{0x034e, 0x09},
-	{0x034f, 0xa0},
-	{0x3261, 0x00},
-	{0x3264, 0x00},
-	{0x3265, 0x10},
-	{0x0301, 0x06},
-	{0x0303, 0x04},
-	{0x0305, 0x03},
-	{0x0306, 0x01},
-	{0x0307, 0x2c},
-	{0x0309, 0x0a},
-	{0x030b, 0x02},
-	{0x030d, 0x04},
-	{0x030e, 0x01},
-	{0x030f, 0x2c},
-	{0x0310, 0x01},
-	{0x0820, 0x0b},
-	{0x0821, 0x40},
-	{0x0822, 0x00},
-	{0x0823, 0x00},
-	{0x3e20, 0x01},
-	{0x3e37, 0x00},
-	{0x3e3b, 0x01},
-	{0x38a3, 0x01},
-	{0x38a8, 0x00},
-	{0x38a9, 0x00},
-	{0x38aa, 0x00},
-	{0x38ab, 0x00},
-	{0x3234, 0x00},
-	{0x3fc1, 0x00},
-	{0x3235, 0x00},
-	{0x3802, 0x00},
-	{0x3143, 0x04},
-	{0x360a, 0x00},
-	{0x0b00, 0x00},
-	{0x0106, 0x00},
-	{0x0b05, 0x01},
-	{0x0b06, 0x01},
-	{0x3230, 0x00},
-	{0x3602, 0x01},
-	{0x3607, 0x01},
-	{0x3c00, 0x00},
-	{0x3c01, 0x48},
-	{0x3c02, 0xc8},
-	{0x3c03, 0xaa},
-	{0x3c04, 0x91},
-	{0x3c05, 0x54},
-	{0x3c06, 0x26},
-	{0x3c07, 0x20},
-	{0x3c08, 0x51},
-	{0x3d80, 0x00},
-	{0x3f50, 0x00},
-	{0x3f56, 0x00},
-	{0x3f57, 0x30},
-	{0x3f78, 0x01},
-	{0x3f79, 0x18},
-	{0x3f7c, 0x00},
-	{0x3f7d, 0x00},
-	{0x3fba, 0x00},
-	{0x3fbb, 0x00},
-	{0xa081, 0x00},
-	{0xe014, 0x00},
-	{0x0202, 0x0a},
-	{0x0203, 0x66},
-	{0x0224, 0x01},
-	{0x0225, 0xf4},
-	{0x0204, 0x00},
-	{0x0205, 0x00},
-	{0x0216, 0x00},
-	{0x0217, 0x00},
-	{0x020e, 0x01},
-	{0x020f, 0x00},
-	{0x0210, 0x01},
-	{0x0211, 0x00},
-	{0x0212, 0x01},
-	{0x0213, 0x00},
-	{0x0214, 0x01},
-	{0x0215, 0x00},
-	{0x0218, 0x01},
-	{0x0219, 0x00},
-	{0x3614, 0x00},
-	{0x3616, 0x0d},
-	{0x3617, 0x56},
-	{0xb612, 0x20},
-	{0xb613, 0x20},
-	{0xb614, 0x20},
-	{0xb615, 0x20},
-	{0xb616, 0x0a},
-	{0xb617, 0x0a},
-	{0xb618, 0x20},
-	{0xb619, 0x20},
-	{0xb61a, 0x20},
-	{0xb61b, 0x20},
-	{0xb61c, 0x0a},
-	{0xb61d, 0x0a},
-	{0xb666, 0x30},
-	{0xb667, 0x30},
-	{0xb668, 0x30},
-	{0xb669, 0x30},
-	{0xb66a, 0x14},
-	{0xb66b, 0x14},
-	{0xb66c, 0x20},
-	{0xb66d, 0x20},
-	{0xb66e, 0x20},
-	{0xb66f, 0x20},
-	{0xb670, 0x10},
-	{0xb671, 0x10},
-	{0x3237, 0x00},
-	{0x3900, 0x00},
-	{0x3901, 0x00},
-	{0x3902, 0x00},
-	{0x3904, 0x00},
-	{0x3905, 0x00},
-	{0x3906, 0x00},
-	{0x3907, 0x00},
-	{0x3908, 0x00},
-	{0x3909, 0x00},
-	{0x3912, 0x00},
-	{0x3930, 0x00},
-	{0x3931, 0x00},
-	{0x3933, 0x00},
-	{0x3934, 0x00},
-	{0x3935, 0x00},
-	{0x3936, 0x00},
-	{0x3937, 0x00},
-	{0x30ac, 0x00},
+	{ 0x0112, 0x0a },
+	{ 0x0113, 0x0a },
+	{ 0x0114, 0x03 },
+	{ 0x0342, 0x0f },
+	{ 0x0343, 0x80 },
+	{ 0x0340, 0x0a },
+	{ 0x0341, 0x78 },
+	{ 0x0344, 0x00 },
+	{ 0x0345, 0x00 },
+	{ 0x0346, 0x00 },
+	{ 0x0347, 0x00 },
+	{ 0x0348, 0x0c },
+	{ 0x0349, 0xcf },
+	{ 0x034a, 0x09 },
+	{ 0x034b, 0x9f },
+	{ 0x0220, 0x00 },
+	{ 0x0221, 0x11 },
+	{ 0x0381, 0x01 },
+	{ 0x0383, 0x01 },
+	{ 0x0385, 0x01 },
+	{ 0x0387, 0x01 },
+	{ 0x0900, 0x00 },
+	{ 0x0901, 0x11 },
+	{ 0x0902, 0x0a },
+	{ 0x3140, 0x02 },
+	{ 0x3141, 0x00 },
+	{ 0x3f0d, 0x0a },
+	{ 0x3f14, 0x01 },
+	{ 0x3f3c, 0x01 },
+	{ 0x3f4d, 0x01 },
+	{ 0x3f4c, 0x01 },
+	{ 0x4254, 0x7f },
+	{ 0x0401, 0x00 },
+	{ 0x0404, 0x00 },
+	{ 0x0405, 0x10 },
+	{ 0x0408, 0x00 },
+	{ 0x0409, 0x00 },
+	{ 0x040a, 0x00 },
+	{ 0x040b, 0x00 },
+	{ 0x040c, 0x0c },
+	{ 0x040d, 0xd0 },
+	{ 0x040e, 0x09 },
+	{ 0x040f, 0xa0 },
+	{ 0x034c, 0x0c },
+	{ 0x034d, 0xd0 },
+	{ 0x034e, 0x09 },
+	{ 0x034f, 0xa0 },
+	{ 0x3261, 0x00 },
+	{ 0x3264, 0x00 },
+	{ 0x3265, 0x10 },
+	{ 0x0301, 0x06 },
+	{ 0x0303, 0x04 },
+	{ 0x0305, 0x03 },
+	{ 0x0306, 0x01 },
+	{ 0x0307, 0x2c },
+	{ 0x0309, 0x0a },
+	{ 0x030b, 0x02 },
+	{ 0x030d, 0x04 },
+	{ 0x030e, 0x01 },
+	{ 0x030f, 0x2c },
+	{ 0x0310, 0x01 },
+	{ 0x0820, 0x0b },
+	{ 0x0821, 0x40 },
+	{ 0x0822, 0x00 },
+	{ 0x0823, 0x00 },
+	{ 0x3e20, 0x01 },
+	{ 0x3e37, 0x00 },
+	{ 0x3e3b, 0x01 },
+	{ 0x38a3, 0x01 },
+	{ 0x38a8, 0x00 },
+	{ 0x38a9, 0x00 },
+	{ 0x38aa, 0x00 },
+	{ 0x38ab, 0x00 },
+	{ 0x3234, 0x00 },
+	{ 0x3fc1, 0x00 },
+	{ 0x3235, 0x00 },
+	{ 0x3802, 0x00 },
+	{ 0x3143, 0x04 },
+	{ 0x360a, 0x00 },
+	{ 0x0b00, 0x00 },
+	{ 0x0106, 0x00 },
+	{ 0x0b05, 0x01 },
+	{ 0x0b06, 0x01 },
+	{ 0x3230, 0x00 },
+	{ 0x3602, 0x01 },
+	{ 0x3607, 0x01 },
+	{ 0x3c00, 0x00 },
+	{ 0x3c01, 0x48 },
+	{ 0x3c02, 0xc8 },
+	{ 0x3c03, 0xaa },
+	{ 0x3c04, 0x91 },
+	{ 0x3c05, 0x54 },
+	{ 0x3c06, 0x26 },
+	{ 0x3c07, 0x20 },
+	{ 0x3c08, 0x51 },
+	{ 0x3d80, 0x00 },
+	{ 0x3f50, 0x00 },
+	{ 0x3f56, 0x00 },
+	{ 0x3f57, 0x30 },
+	{ 0x3f78, 0x01 },
+	{ 0x3f79, 0x18 },
+	{ 0x3f7c, 0x00 },
+	{ 0x3f7d, 0x00 },
+	{ 0x3fba, 0x00 },
+	{ 0x3fbb, 0x00 },
+	{ 0xa081, 0x00 },
+	{ 0xe014, 0x00 },
+	{ 0x0202, 0x0a },
+	{ 0x0203, 0x66 },
+	{ 0x0224, 0x01 },
+	{ 0x0225, 0xf4 },
+	{ 0x0204, 0x00 },
+	{ 0x0205, 0x00 },
+	{ 0x0216, 0x00 },
+	{ 0x0217, 0x00 },
+	{ 0x020e, 0x01 },
+	{ 0x020f, 0x00 },
+	{ 0x0210, 0x01 },
+	{ 0x0211, 0x00 },
+	{ 0x0212, 0x01 },
+	{ 0x0213, 0x00 },
+	{ 0x0214, 0x01 },
+	{ 0x0215, 0x00 },
+	{ 0x0218, 0x01 },
+	{ 0x0219, 0x00 },
+	{ 0x3614, 0x00 },
+	{ 0x3616, 0x0d },
+	{ 0x3617, 0x56 },
+	{ 0xb612, 0x20 },
+	{ 0xb613, 0x20 },
+	{ 0xb614, 0x20 },
+	{ 0xb615, 0x20 },
+	{ 0xb616, 0x0a },
+	{ 0xb617, 0x0a },
+	{ 0xb618, 0x20 },
+	{ 0xb619, 0x20 },
+	{ 0xb61a, 0x20 },
+	{ 0xb61b, 0x20 },
+	{ 0xb61c, 0x0a },
+	{ 0xb61d, 0x0a },
+	{ 0xb666, 0x30 },
+	{ 0xb667, 0x30 },
+	{ 0xb668, 0x30 },
+	{ 0xb669, 0x30 },
+	{ 0xb66a, 0x14 },
+	{ 0xb66b, 0x14 },
+	{ 0xb66c, 0x20 },
+	{ 0xb66d, 0x20 },
+	{ 0xb66e, 0x20 },
+	{ 0xb66f, 0x20 },
+	{ 0xb670, 0x10 },
+	{ 0xb671, 0x10 },
+	{ 0x3237, 0x00 },
+	{ 0x3900, 0x00 },
+	{ 0x3901, 0x00 },
+	{ 0x3902, 0x00 },
+	{ 0x3904, 0x00 },
+	{ 0x3905, 0x00 },
+	{ 0x3906, 0x00 },
+	{ 0x3907, 0x00 },
+	{ 0x3908, 0x00 },
+	{ 0x3909, 0x00 },
+	{ 0x3912, 0x00 },
+	{ 0x3930, 0x00 },
+	{ 0x3931, 0x00 },
+	{ 0x3933, 0x00 },
+	{ 0x3934, 0x00 },
+	{ 0x3935, 0x00 },
+	{ 0x3936, 0x00 },
+	{ 0x3937, 0x00 },
+	{ 0x30ac, 0x00 },
 };
 
 static const struct imx319_reg mode_1936x1096_regs[] = {
-	{0x0112, 0x0a},
-	{0x0113, 0x0a},
-	{0x0114, 0x03},
-	{0x0342, 0x0f},
-	{0x0343, 0x80},
-	{0x0340, 0x07},
-	{0x0341, 0x72},
-	{0x0344, 0x00},
-	{0x0345, 0x00},
-	{0x0346, 0x01},
-	{0x0347, 0x30},
-	{0x0348, 0x0c},
-	{0x0349, 0xcf},
-	{0x034a, 0x08},
-	{0x034b, 0x6f},
-	{0x0220, 0x00},
-	{0x0221, 0x11},
-	{0x0381, 0x01},
-	{0x0383, 0x01},
-	{0x0385, 0x01},
-	{0x0387, 0x01},
-	{0x0900, 0x00},
-	{0x0901, 0x11},
-	{0x0902, 0x0a},
-	{0x3140, 0x02},
-	{0x3141, 0x00},
-	{0x3f0d, 0x0a},
-	{0x3f14, 0x01},
-	{0x3f3c, 0x01},
-	{0x3f4d, 0x01},
-	{0x3f4c, 0x01},
-	{0x4254, 0x7f},
-	{0x0401, 0x00},
-	{0x0404, 0x00},
-	{0x0405, 0x10},
-	{0x0408, 0x02},
-	{0x0409, 0xa0},
-	{0x040a, 0x01},
-	{0x040b, 0x7c},
-	{0x040c, 0x07},
-	{0x040d, 0x90},
-	{0x040e, 0x04},
-	{0x040f, 0x48},
-	{0x034c, 0x07},
-	{0x034d, 0x90},
-	{0x034e, 0x04},
-	{0x034f, 0x48},
-	{0x3261, 0x00},
-	{0x3264, 0x00},
-	{0x3265, 0x10},
-	{0x0301, 0x06},
-	{0x0303, 0x02},
-	{0x0305, 0x03},
-	{0x0306, 0x00},
-	{0x0307, 0xd6},
-	{0x0309, 0x0a},
-	{0x030b, 0x02},
-	{0x030d, 0x04},
-	{0x030e, 0x01},
-	{0x030f, 0x2c},
-	{0x0310, 0x01},
-	{0x0820, 0x0b},
-	{0x0821, 0x40},
-	{0x0822, 0x00},
-	{0x0823, 0x00},
-	{0x3e20, 0x01},
-	{0x3e37, 0x00},
-	{0x3e3b, 0x01},
-	{0x38a3, 0x01},
-	{0x38a8, 0x00},
-	{0x38a9, 0x00},
-	{0x38aa, 0x00},
-	{0x38ab, 0x00},
-	{0x3234, 0x00},
-	{0x3fc1, 0x00},
-	{0x3235, 0x00},
-	{0x3802, 0x00},
-	{0x3143, 0x04},
-	{0x360a, 0x00},
-	{0x0b00, 0x00},
-	{0x0106, 0x00},
-	{0x0b05, 0x01},
-	{0x0b06, 0x01},
-	{0x3230, 0x00},
-	{0x3602, 0x01},
-	{0x3607, 0x01},
-	{0x3c00, 0x00},
-	{0x3c01, 0x48},
-	{0x3c02, 0xc8},
-	{0x3c03, 0xaa},
-	{0x3c04, 0x91},
-	{0x3c05, 0x54},
-	{0x3c06, 0x26},
-	{0x3c07, 0x20},
-	{0x3c08, 0x51},
-	{0x3d80, 0x00},
-	{0x3f50, 0x00},
-	{0x3f56, 0x00},
-	{0x3f57, 0x30},
-	{0x3f78, 0x01},
-	{0x3f79, 0x18},
-	{0x3f7c, 0x00},
-	{0x3f7d, 0x00},
-	{0x3fba, 0x00},
-	{0x3fbb, 0x00},
-	{0xa081, 0x00},
-	{0xe014, 0x00},
-	{0x0202, 0x07},
-	{0x0203, 0x60},
-	{0x0224, 0x01},
-	{0x0225, 0xf4},
-	{0x0204, 0x00},
-	{0x0205, 0x00},
-	{0x0216, 0x00},
-	{0x0217, 0x00},
-	{0x020e, 0x01},
-	{0x020f, 0x00},
-	{0x0210, 0x01},
-	{0x0211, 0x00},
-	{0x0212, 0x01},
-	{0x0213, 0x00},
-	{0x0214, 0x01},
-	{0x0215, 0x00},
-	{0x0218, 0x01},
-	{0x0219, 0x00},
-	{0x3614, 0x00},
-	{0x3616, 0x0d},
-	{0x3617, 0x56},
-	{0xb612, 0x20},
-	{0xb613, 0x20},
-	{0xb614, 0x20},
-	{0xb615, 0x20},
-	{0xb616, 0x0a},
-	{0xb617, 0x0a},
-	{0xb618, 0x20},
-	{0xb619, 0x20},
-	{0xb61a, 0x20},
-	{0xb61b, 0x20},
-	{0xb61c, 0x0a},
-	{0xb61d, 0x0a},
-	{0xb666, 0x30},
-	{0xb667, 0x30},
-	{0xb668, 0x30},
-	{0xb669, 0x30},
-	{0xb66a, 0x14},
-	{0xb66b, 0x14},
-	{0xb66c, 0x20},
-	{0xb66d, 0x20},
-	{0xb66e, 0x20},
-	{0xb66f, 0x20},
-	{0xb670, 0x10},
-	{0xb671, 0x10},
-	{0x3237, 0x00},
-	{0x3900, 0x00},
-	{0x3901, 0x00},
-	{0x3902, 0x00},
-	{0x3904, 0x00},
-	{0x3905, 0x00},
-	{0x3906, 0x00},
-	{0x3907, 0x00},
-	{0x3908, 0x00},
-	{0x3909, 0x00},
-	{0x3912, 0x00},
-	{0x3930, 0x00},
-	{0x3931, 0x00},
-	{0x3933, 0x00},
-	{0x3934, 0x00},
-	{0x3935, 0x00},
-	{0x3936, 0x00},
-	{0x3937, 0x00},
-	{0x30ac, 0x00},
+	{ 0x0112, 0x0a },
+	{ 0x0113, 0x0a },
+	{ 0x0114, 0x03 },
+	{ 0x0342, 0x0f },
+	{ 0x0343, 0x80 },
+	{ 0x0340, 0x07 },
+	{ 0x0341, 0x72 },
+	{ 0x0344, 0x00 },
+	{ 0x0345, 0x00 },
+	{ 0x0346, 0x01 },
+	{ 0x0347, 0x30 },
+	{ 0x0348, 0x0c },
+	{ 0x0349, 0xcf },
+	{ 0x034a, 0x08 },
+	{ 0x034b, 0x6f },
+	{ 0x0220, 0x00 },
+	{ 0x0221, 0x11 },
+	{ 0x0381, 0x01 },
+	{ 0x0383, 0x01 },
+	{ 0x0385, 0x01 },
+	{ 0x0387, 0x01 },
+	{ 0x0900, 0x00 },
+	{ 0x0901, 0x11 },
+	{ 0x0902, 0x0a },
+	{ 0x3140, 0x02 },
+	{ 0x3141, 0x00 },
+	{ 0x3f0d, 0x0a },
+	{ 0x3f14, 0x01 },
+	{ 0x3f3c, 0x01 },
+	{ 0x3f4d, 0x01 },
+	{ 0x3f4c, 0x01 },
+	{ 0x4254, 0x7f },
+	{ 0x0401, 0x00 },
+	{ 0x0404, 0x00 },
+	{ 0x0405, 0x10 },
+	{ 0x0408, 0x02 },
+	{ 0x0409, 0xa0 },
+	{ 0x040a, 0x01 },
+	{ 0x040b, 0x7c },
+	{ 0x040c, 0x07 },
+	{ 0x040d, 0x90 },
+	{ 0x040e, 0x04 },
+	{ 0x040f, 0x48 },
+	{ 0x034c, 0x07 },
+	{ 0x034d, 0x90 },
+	{ 0x034e, 0x04 },
+	{ 0x034f, 0x48 },
+	{ 0x3261, 0x00 },
+	{ 0x3264, 0x00 },
+	{ 0x3265, 0x10 },
+	{ 0x0301, 0x06 },
+	{ 0x0303, 0x02 },
+	{ 0x0305, 0x03 },
+	{ 0x0306, 0x00 },
+	{ 0x0307, 0xd6 },
+	{ 0x0309, 0x0a },
+	{ 0x030b, 0x02 },
+	{ 0x030d, 0x04 },
+	{ 0x030e, 0x01 },
+	{ 0x030f, 0x2c },
+	{ 0x0310, 0x01 },
+	{ 0x0820, 0x0b },
+	{ 0x0821, 0x40 },
+	{ 0x0822, 0x00 },
+	{ 0x0823, 0x00 },
+	{ 0x3e20, 0x01 },
+	{ 0x3e37, 0x00 },
+	{ 0x3e3b, 0x01 },
+	{ 0x38a3, 0x01 },
+	{ 0x38a8, 0x00 },
+	{ 0x38a9, 0x00 },
+	{ 0x38aa, 0x00 },
+	{ 0x38ab, 0x00 },
+	{ 0x3234, 0x00 },
+	{ 0x3fc1, 0x00 },
+	{ 0x3235, 0x00 },
+	{ 0x3802, 0x00 },
+	{ 0x3143, 0x04 },
+	{ 0x360a, 0x00 },
+	{ 0x0b00, 0x00 },
+	{ 0x0106, 0x00 },
+	{ 0x0b05, 0x01 },
+	{ 0x0b06, 0x01 },
+	{ 0x3230, 0x00 },
+	{ 0x3602, 0x01 },
+	{ 0x3607, 0x01 },
+	{ 0x3c00, 0x00 },
+	{ 0x3c01, 0x48 },
+	{ 0x3c02, 0xc8 },
+	{ 0x3c03, 0xaa },
+	{ 0x3c04, 0x91 },
+	{ 0x3c05, 0x54 },
+	{ 0x3c06, 0x26 },
+	{ 0x3c07, 0x20 },
+	{ 0x3c08, 0x51 },
+	{ 0x3d80, 0x00 },
+	{ 0x3f50, 0x00 },
+	{ 0x3f56, 0x00 },
+	{ 0x3f57, 0x30 },
+	{ 0x3f78, 0x01 },
+	{ 0x3f79, 0x18 },
+	{ 0x3f7c, 0x00 },
+	{ 0x3f7d, 0x00 },
+	{ 0x3fba, 0x00 },
+	{ 0x3fbb, 0x00 },
+	{ 0xa081, 0x00 },
+	{ 0xe014, 0x00 },
+	{ 0x0202, 0x07 },
+	{ 0x0203, 0x60 },
+	{ 0x0224, 0x01 },
+	{ 0x0225, 0xf4 },
+	{ 0x0204, 0x00 },
+	{ 0x0205, 0x00 },
+	{ 0x0216, 0x00 },
+	{ 0x0217, 0x00 },
+	{ 0x020e, 0x01 },
+	{ 0x020f, 0x00 },
+	{ 0x0210, 0x01 },
+	{ 0x0211, 0x00 },
+	{ 0x0212, 0x01 },
+	{ 0x0213, 0x00 },
+	{ 0x0214, 0x01 },
+	{ 0x0215, 0x00 },
+	{ 0x0218, 0x01 },
+	{ 0x0219, 0x00 },
+	{ 0x3614, 0x00 },
+	{ 0x3616, 0x0d },
+	{ 0x3617, 0x56 },
+	{ 0xb612, 0x20 },
+	{ 0xb613, 0x20 },
+	{ 0xb614, 0x20 },
+	{ 0xb615, 0x20 },
+	{ 0xb616, 0x0a },
+	{ 0xb617, 0x0a },
+	{ 0xb618, 0x20 },
+	{ 0xb619, 0x20 },
+	{ 0xb61a, 0x20 },
+	{ 0xb61b, 0x20 },
+	{ 0xb61c, 0x0a },
+	{ 0xb61d, 0x0a },
+	{ 0xb666, 0x30 },
+	{ 0xb667, 0x30 },
+	{ 0xb668, 0x30 },
+	{ 0xb669, 0x30 },
+	{ 0xb66a, 0x14 },
+	{ 0xb66b, 0x14 },
+	{ 0xb66c, 0x20 },
+	{ 0xb66d, 0x20 },
+	{ 0xb66e, 0x20 },
+	{ 0xb66f, 0x20 },
+	{ 0xb670, 0x10 },
+	{ 0xb671, 0x10 },
+	{ 0x3237, 0x00 },
+	{ 0x3900, 0x00 },
+	{ 0x3901, 0x00 },
+	{ 0x3902, 0x00 },
+	{ 0x3904, 0x00 },
+	{ 0x3905, 0x00 },
+	{ 0x3906, 0x00 },
+	{ 0x3907, 0x00 },
+	{ 0x3908, 0x00 },
+	{ 0x3909, 0x00 },
+	{ 0x3912, 0x00 },
+	{ 0x3930, 0x00 },
+	{ 0x3931, 0x00 },
+	{ 0x3933, 0x00 },
+	{ 0x3934, 0x00 },
+	{ 0x3935, 0x00 },
+	{ 0x3936, 0x00 },
+	{ 0x3937, 0x00 },
+	{ 0x30ac, 0x00 },
 };
 
 static const struct imx319_reg mode_1920x1080_regs[] = {
-	{0x0112, 0x0a},
-	{0x0113, 0x0a},
-	{0x0114, 0x03},
-	{0x0342, 0x0f},
-	{0x0343, 0x80},
-	{0x0340, 0x07},
-	{0x0341, 0x72},
-	{0x0344, 0x00},
-	{0x0345, 0x00},
-	{0x0346, 0x01},
-	{0x0347, 0x30},
-	{0x0348, 0x0c},
-	{0x0349, 0xcf},
-	{0x034a, 0x08},
-	{0x034b, 0x6f},
-	{0x0220, 0x00},
-	{0x0221, 0x11},
-	{0x0381, 0x01},
-	{0x0383, 0x01},
-	{0x0385, 0x01},
-	{0x0387, 0x01},
-	{0x0900, 0x00},
-	{0x0901, 0x11},
-	{0x0902, 0x0a},
-	{0x3140, 0x02},
-	{0x3141, 0x00},
-	{0x3f0d, 0x0a},
-	{0x3f14, 0x01},
-	{0x3f3c, 0x01},
-	{0x3f4d, 0x01},
-	{0x3f4c, 0x01},
-	{0x4254, 0x7f},
-	{0x0401, 0x00},
-	{0x0404, 0x00},
-	{0x0405, 0x10},
-	{0x0408, 0x02},
-	{0x0409, 0xa8},
-	{0x040a, 0x01},
-	{0x040b, 0x84},
-	{0x040c, 0x07},
-	{0x040d, 0x80},
-	{0x040e, 0x04},
-	{0x040f, 0x38},
-	{0x034c, 0x07},
-	{0x034d, 0x80},
-	{0x034e, 0x04},
-	{0x034f, 0x38},
-	{0x3261, 0x00},
-	{0x3264, 0x00},
-	{0x3265, 0x10},
-	{0x0301, 0x06},
-	{0x0303, 0x02},
-	{0x0305, 0x03},
-	{0x0306, 0x00},
-	{0x0307, 0xd6},
-	{0x0309, 0x0a},
-	{0x030b, 0x02},
-	{0x030d, 0x04},
-	{0x030e, 0x01},
-	{0x030f, 0x2c},
-	{0x0310, 0x01},
-	{0x0820, 0x0b},
-	{0x0821, 0x40},
-	{0x0822, 0x00},
-	{0x0823, 0x00},
-	{0x3e20, 0x01},
-	{0x3e37, 0x00},
-	{0x3e3b, 0x01},
-	{0x38a3, 0x01},
-	{0x38a8, 0x00},
-	{0x38a9, 0x00},
-	{0x38aa, 0x00},
-	{0x38ab, 0x00},
-	{0x3234, 0x00},
-	{0x3fc1, 0x00},
-	{0x3235, 0x00},
-	{0x3802, 0x00},
-	{0x3143, 0x04},
-	{0x360a, 0x00},
-	{0x0b00, 0x00},
-	{0x0106, 0x00},
-	{0x0b05, 0x01},
-	{0x0b06, 0x01},
-	{0x3230, 0x00},
-	{0x3602, 0x01},
-	{0x3607, 0x01},
-	{0x3c00, 0x00},
-	{0x3c01, 0x48},
-	{0x3c02, 0xc8},
-	{0x3c03, 0xaa},
-	{0x3c04, 0x91},
-	{0x3c05, 0x54},
-	{0x3c06, 0x26},
-	{0x3c07, 0x20},
-	{0x3c08, 0x51},
-	{0x3d80, 0x00},
-	{0x3f50, 0x00},
-	{0x3f56, 0x00},
-	{0x3f57, 0x30},
-	{0x3f78, 0x01},
-	{0x3f79, 0x18},
-	{0x3f7c, 0x00},
-	{0x3f7d, 0x00},
-	{0x3fba, 0x00},
-	{0x3fbb, 0x00},
-	{0xa081, 0x00},
-	{0xe014, 0x00},
-	{0x0202, 0x07},
-	{0x0203, 0x60},
-	{0x0224, 0x01},
-	{0x0225, 0xf4},
-	{0x0204, 0x00},
-	{0x0205, 0x00},
-	{0x0216, 0x00},
-	{0x0217, 0x00},
-	{0x020e, 0x01},
-	{0x020f, 0x00},
-	{0x0210, 0x01},
-	{0x0211, 0x00},
-	{0x0212, 0x01},
-	{0x0213, 0x00},
-	{0x0214, 0x01},
-	{0x0215, 0x00},
-	{0x0218, 0x01},
-	{0x0219, 0x00},
-	{0x3614, 0x00},
-	{0x3616, 0x0d},
-	{0x3617, 0x56},
-	{0xb612, 0x20},
-	{0xb613, 0x20},
-	{0xb614, 0x20},
-	{0xb615, 0x20},
-	{0xb616, 0x0a},
-	{0xb617, 0x0a},
-	{0xb618, 0x20},
-	{0xb619, 0x20},
-	{0xb61a, 0x20},
-	{0xb61b, 0x20},
-	{0xb61c, 0x0a},
-	{0xb61d, 0x0a},
-	{0xb666, 0x30},
-	{0xb667, 0x30},
-	{0xb668, 0x30},
-	{0xb669, 0x30},
-	{0xb66a, 0x14},
-	{0xb66b, 0x14},
-	{0xb66c, 0x20},
-	{0xb66d, 0x20},
-	{0xb66e, 0x20},
-	{0xb66f, 0x20},
-	{0xb670, 0x10},
-	{0xb671, 0x10},
-	{0x3237, 0x00},
-	{0x3900, 0x00},
-	{0x3901, 0x00},
-	{0x3902, 0x00},
-	{0x3904, 0x00},
-	{0x3905, 0x00},
-	{0x3906, 0x00},
-	{0x3907, 0x00},
-	{0x3908, 0x00},
-	{0x3909, 0x00},
-	{0x3912, 0x00},
-	{0x3930, 0x00},
-	{0x3931, 0x00},
-	{0x3933, 0x00},
-	{0x3934, 0x00},
-	{0x3935, 0x00},
-	{0x3936, 0x00},
-	{0x3937, 0x00},
-	{0x30ac, 0x00},
+	{ 0x0112, 0x0a },
+	{ 0x0113, 0x0a },
+	{ 0x0114, 0x03 },
+	{ 0x0342, 0x0f },
+	{ 0x0343, 0x80 },
+	{ 0x0340, 0x07 },
+	{ 0x0341, 0x72 },
+	{ 0x0344, 0x00 },
+	{ 0x0345, 0x00 },
+	{ 0x0346, 0x01 },
+	{ 0x0347, 0x30 },
+	{ 0x0348, 0x0c },
+	{ 0x0349, 0xcf },
+	{ 0x034a, 0x08 },
+	{ 0x034b, 0x6f },
+	{ 0x0220, 0x00 },
+	{ 0x0221, 0x11 },
+	{ 0x0381, 0x01 },
+	{ 0x0383, 0x01 },
+	{ 0x0385, 0x01 },
+	{ 0x0387, 0x01 },
+	{ 0x0900, 0x00 },
+	{ 0x0901, 0x11 },
+	{ 0x0902, 0x0a },
+	{ 0x3140, 0x02 },
+	{ 0x3141, 0x00 },
+	{ 0x3f0d, 0x0a },
+	{ 0x3f14, 0x01 },
+	{ 0x3f3c, 0x01 },
+	{ 0x3f4d, 0x01 },
+	{ 0x3f4c, 0x01 },
+	{ 0x4254, 0x7f },
+	{ 0x0401, 0x00 },
+	{ 0x0404, 0x00 },
+	{ 0x0405, 0x10 },
+	{ 0x0408, 0x02 },
+	{ 0x0409, 0xa8 },
+	{ 0x040a, 0x01 },
+	{ 0x040b, 0x84 },
+	{ 0x040c, 0x07 },
+	{ 0x040d, 0x80 },
+	{ 0x040e, 0x04 },
+	{ 0x040f, 0x38 },
+	{ 0x034c, 0x07 },
+	{ 0x034d, 0x80 },
+	{ 0x034e, 0x04 },
+	{ 0x034f, 0x38 },
+	{ 0x3261, 0x00 },
+	{ 0x3264, 0x00 },
+	{ 0x3265, 0x10 },
+	{ 0x0301, 0x06 },
+	{ 0x0303, 0x02 },
+	{ 0x0305, 0x03 },
+	{ 0x0306, 0x00 },
+	{ 0x0307, 0xd6 },
+	{ 0x0309, 0x0a },
+	{ 0x030b, 0x02 },
+	{ 0x030d, 0x04 },
+	{ 0x030e, 0x01 },
+	{ 0x030f, 0x2c },
+	{ 0x0310, 0x01 },
+	{ 0x0820, 0x0b },
+	{ 0x0821, 0x40 },
+	{ 0x0822, 0x00 },
+	{ 0x0823, 0x00 },
+	{ 0x3e20, 0x01 },
+	{ 0x3e37, 0x00 },
+	{ 0x3e3b, 0x01 },
+	{ 0x38a3, 0x01 },
+	{ 0x38a8, 0x00 },
+	{ 0x38a9, 0x00 },
+	{ 0x38aa, 0x00 },
+	{ 0x38ab, 0x00 },
+	{ 0x3234, 0x00 },
+	{ 0x3fc1, 0x00 },
+	{ 0x3235, 0x00 },
+	{ 0x3802, 0x00 },
+	{ 0x3143, 0x04 },
+	{ 0x360a, 0x00 },
+	{ 0x0b00, 0x00 },
+	{ 0x0106, 0x00 },
+	{ 0x0b05, 0x01 },
+	{ 0x0b06, 0x01 },
+	{ 0x3230, 0x00 },
+	{ 0x3602, 0x01 },
+	{ 0x3607, 0x01 },
+	{ 0x3c00, 0x00 },
+	{ 0x3c01, 0x48 },
+	{ 0x3c02, 0xc8 },
+	{ 0x3c03, 0xaa },
+	{ 0x3c04, 0x91 },
+	{ 0x3c05, 0x54 },
+	{ 0x3c06, 0x26 },
+	{ 0x3c07, 0x20 },
+	{ 0x3c08, 0x51 },
+	{ 0x3d80, 0x00 },
+	{ 0x3f50, 0x00 },
+	{ 0x3f56, 0x00 },
+	{ 0x3f57, 0x30 },
+	{ 0x3f78, 0x01 },
+	{ 0x3f79, 0x18 },
+	{ 0x3f7c, 0x00 },
+	{ 0x3f7d, 0x00 },
+	{ 0x3fba, 0x00 },
+	{ 0x3fbb, 0x00 },
+	{ 0xa081, 0x00 },
+	{ 0xe014, 0x00 },
+	{ 0x0202, 0x07 },
+	{ 0x0203, 0x60 },
+	{ 0x0224, 0x01 },
+	{ 0x0225, 0xf4 },
+	{ 0x0204, 0x00 },
+	{ 0x0205, 0x00 },
+	{ 0x0216, 0x00 },
+	{ 0x0217, 0x00 },
+	{ 0x020e, 0x01 },
+	{ 0x020f, 0x00 },
+	{ 0x0210, 0x01 },
+	{ 0x0211, 0x00 },
+	{ 0x0212, 0x01 },
+	{ 0x0213, 0x00 },
+	{ 0x0214, 0x01 },
+	{ 0x0215, 0x00 },
+	{ 0x0218, 0x01 },
+	{ 0x0219, 0x00 },
+	{ 0x3614, 0x00 },
+	{ 0x3616, 0x0d },
+	{ 0x3617, 0x56 },
+	{ 0xb612, 0x20 },
+	{ 0xb613, 0x20 },
+	{ 0xb614, 0x20 },
+	{ 0xb615, 0x20 },
+	{ 0xb616, 0x0a },
+	{ 0xb617, 0x0a },
+	{ 0xb618, 0x20 },
+	{ 0xb619, 0x20 },
+	{ 0xb61a, 0x20 },
+	{ 0xb61b, 0x20 },
+	{ 0xb61c, 0x0a },
+	{ 0xb61d, 0x0a },
+	{ 0xb666, 0x30 },
+	{ 0xb667, 0x30 },
+	{ 0xb668, 0x30 },
+	{ 0xb669, 0x30 },
+	{ 0xb66a, 0x14 },
+	{ 0xb66b, 0x14 },
+	{ 0xb66c, 0x20 },
+	{ 0xb66d, 0x20 },
+	{ 0xb66e, 0x20 },
+	{ 0xb66f, 0x20 },
+	{ 0xb670, 0x10 },
+	{ 0xb671, 0x10 },
+	{ 0x3237, 0x00 },
+	{ 0x3900, 0x00 },
+	{ 0x3901, 0x00 },
+	{ 0x3902, 0x00 },
+	{ 0x3904, 0x00 },
+	{ 0x3905, 0x00 },
+	{ 0x3906, 0x00 },
+	{ 0x3907, 0x00 },
+	{ 0x3908, 0x00 },
+	{ 0x3909, 0x00 },
+	{ 0x3912, 0x00 },
+	{ 0x3930, 0x00 },
+	{ 0x3931, 0x00 },
+	{ 0x3933, 0x00 },
+	{ 0x3934, 0x00 },
+	{ 0x3935, 0x00 },
+	{ 0x3936, 0x00 },
+	{ 0x3937, 0x00 },
+	{ 0x30ac, 0x00 },
 };
 
 static const struct imx319_reg mode_1640x1232_regs[] = {
-	{0x0112, 0x0a},
-	{0x0113, 0x0a},
-	{0x0114, 0x03},
-	{0x0342, 0x08},
-	{0x0343, 0x20},
-	{0x0340, 0x05},
-	{0x0341, 0x28},
-	{0x0344, 0x00},
-	{0x0345, 0x00},
-	{0x0346, 0x00},
-	{0x0347, 0x00},
-	{0x0348, 0x0c},
-	{0x0349, 0xcf},
-	{0x034a, 0x09},
-	{0x034b, 0x9f},
-	{0x0220, 0x00},
-	{0x0221, 0x11},
-	{0x0381, 0x01},
-	{0x0383, 0x01},
-	{0x0385, 0x01},
-	{0x0387, 0x01},
-	{0x0900, 0x01},
-	{0x0901, 0x22},
-	{0x0902, 0x0a},
-	{0x3140, 0x02},
-	{0x3141, 0x00},
-	{0x3f0d, 0x0a},
-	{0x3f14, 0x01},
-	{0x3f3c, 0x02},
-	{0x3f4d, 0x01},
-	{0x3f4c, 0x01},
-	{0x4254, 0x7f},
-	{0x0401, 0x00},
-	{0x0404, 0x00},
-	{0x0405, 0x10},
-	{0x0408, 0x00},
-	{0x0409, 0x00},
-	{0x040a, 0x00},
-	{0x040b, 0x00},
-	{0x040c, 0x06},
-	{0x040d, 0x68},
-	{0x040e, 0x04},
-	{0x040f, 0xd0},
-	{0x034c, 0x06},
-	{0x034d, 0x68},
-	{0x034e, 0x04},
-	{0x034f, 0xd0},
-	{0x3261, 0x00},
-	{0x3264, 0x00},
-	{0x3265, 0x10},
-	{0x0301, 0x06},
-	{0x0303, 0x04},
-	{0x0305, 0x03},
-	{0x0306, 0x01},
-	{0x0307, 0x36},
-	{0x0309, 0x0a},
-	{0x030b, 0x02},
-	{0x030d, 0x04},
-	{0x030e, 0x01},
-	{0x030f, 0x2c},
-	{0x0310, 0x01},
-	{0x0820, 0x0b},
-	{0x0821, 0x40},
-	{0x0822, 0x00},
-	{0x0823, 0x00},
-	{0x3e20, 0x01},
-	{0x3e37, 0x00},
-	{0x3e3b, 0x01},
-	{0x38a3, 0x01},
-	{0x38a8, 0x00},
-	{0x38a9, 0x00},
-	{0x38aa, 0x00},
-	{0x38ab, 0x00},
-	{0x3234, 0x00},
-	{0x3fc1, 0x00},
-	{0x3235, 0x00},
-	{0x3802, 0x00},
-	{0x3143, 0x04},
-	{0x360a, 0x00},
-	{0x0b00, 0x00},
-	{0x0106, 0x00},
-	{0x0b05, 0x01},
-	{0x0b06, 0x01},
-	{0x3230, 0x00},
-	{0x3602, 0x01},
-	{0x3607, 0x01},
-	{0x3c00, 0x00},
-	{0x3c01, 0xba},
-	{0x3c02, 0xc8},
-	{0x3c03, 0xaa},
-	{0x3c04, 0x91},
-	{0x3c05, 0x54},
-	{0x3c06, 0x26},
-	{0x3c07, 0x20},
-	{0x3c08, 0x51},
-	{0x3d80, 0x00},
-	{0x3f50, 0x00},
-	{0x3f56, 0x00},
-	{0x3f57, 0x30},
-	{0x3f78, 0x00},
-	{0x3f79, 0x34},
-	{0x3f7c, 0x00},
-	{0x3f7d, 0x00},
-	{0x3fba, 0x00},
-	{0x3fbb, 0x00},
-	{0xa081, 0x04},
-	{0xe014, 0x00},
-	{0x0202, 0x05},
-	{0x0203, 0x16},
-	{0x0224, 0x01},
-	{0x0225, 0xf4},
-	{0x0204, 0x00},
-	{0x0205, 0x00},
-	{0x0216, 0x00},
-	{0x0217, 0x00},
-	{0x020e, 0x01},
-	{0x020f, 0x00},
-	{0x0210, 0x01},
-	{0x0211, 0x00},
-	{0x0212, 0x01},
-	{0x0213, 0x00},
-	{0x0214, 0x01},
-	{0x0215, 0x00},
-	{0x0218, 0x01},
-	{0x0219, 0x00},
-	{0x3614, 0x00},
-	{0x3616, 0x0d},
-	{0x3617, 0x56},
-	{0xb612, 0x20},
-	{0xb613, 0x20},
-	{0xb614, 0x20},
-	{0xb615, 0x20},
-	{0xb616, 0x0a},
-	{0xb617, 0x0a},
-	{0xb618, 0x20},
-	{0xb619, 0x20},
-	{0xb61a, 0x20},
-	{0xb61b, 0x20},
-	{0xb61c, 0x0a},
-	{0xb61d, 0x0a},
-	{0xb666, 0x30},
-	{0xb667, 0x30},
-	{0xb668, 0x30},
-	{0xb669, 0x30},
-	{0xb66a, 0x14},
-	{0xb66b, 0x14},
-	{0xb66c, 0x20},
-	{0xb66d, 0x20},
-	{0xb66e, 0x20},
-	{0xb66f, 0x20},
-	{0xb670, 0x10},
-	{0xb671, 0x10},
-	{0x3237, 0x00},
-	{0x3900, 0x00},
-	{0x3901, 0x00},
-	{0x3902, 0x00},
-	{0x3904, 0x00},
-	{0x3905, 0x00},
-	{0x3906, 0x00},
-	{0x3907, 0x00},
-	{0x3908, 0x00},
-	{0x3909, 0x00},
-	{0x3912, 0x00},
-	{0x3930, 0x00},
-	{0x3931, 0x00},
-	{0x3933, 0x00},
-	{0x3934, 0x00},
-	{0x3935, 0x00},
-	{0x3936, 0x00},
-	{0x3937, 0x00},
-	{0x30ac, 0x00},
+	{ 0x0112, 0x0a },
+	{ 0x0113, 0x0a },
+	{ 0x0114, 0x03 },
+	{ 0x0342, 0x08 },
+	{ 0x0343, 0x20 },
+	{ 0x0340, 0x05 },
+	{ 0x0341, 0x28 },
+	{ 0x0344, 0x00 },
+	{ 0x0345, 0x00 },
+	{ 0x0346, 0x00 },
+	{ 0x0347, 0x00 },
+	{ 0x0348, 0x0c },
+	{ 0x0349, 0xcf },
+	{ 0x034a, 0x09 },
+	{ 0x034b, 0x9f },
+	{ 0x0220, 0x00 },
+	{ 0x0221, 0x11 },
+	{ 0x0381, 0x01 },
+	{ 0x0383, 0x01 },
+	{ 0x0385, 0x01 },
+	{ 0x0387, 0x01 },
+	{ 0x0900, 0x01 },
+	{ 0x0901, 0x22 },
+	{ 0x0902, 0x0a },
+	{ 0x3140, 0x02 },
+	{ 0x3141, 0x00 },
+	{ 0x3f0d, 0x0a },
+	{ 0x3f14, 0x01 },
+	{ 0x3f3c, 0x02 },
+	{ 0x3f4d, 0x01 },
+	{ 0x3f4c, 0x01 },
+	{ 0x4254, 0x7f },
+	{ 0x0401, 0x00 },
+	{ 0x0404, 0x00 },
+	{ 0x0405, 0x10 },
+	{ 0x0408, 0x00 },
+	{ 0x0409, 0x00 },
+	{ 0x040a, 0x00 },
+	{ 0x040b, 0x00 },
+	{ 0x040c, 0x06 },
+	{ 0x040d, 0x68 },
+	{ 0x040e, 0x04 },
+	{ 0x040f, 0xd0 },
+	{ 0x034c, 0x06 },
+	{ 0x034d, 0x68 },
+	{ 0x034e, 0x04 },
+	{ 0x034f, 0xd0 },
+	{ 0x3261, 0x00 },
+	{ 0x3264, 0x00 },
+	{ 0x3265, 0x10 },
+	{ 0x0301, 0x06 },
+	{ 0x0303, 0x04 },
+	{ 0x0305, 0x03 },
+	{ 0x0306, 0x01 },
+	{ 0x0307, 0x36 },
+	{ 0x0309, 0x0a },
+	{ 0x030b, 0x02 },
+	{ 0x030d, 0x04 },
+	{ 0x030e, 0x01 },
+	{ 0x030f, 0x2c },
+	{ 0x0310, 0x01 },
+	{ 0x0820, 0x0b },
+	{ 0x0821, 0x40 },
+	{ 0x0822, 0x00 },
+	{ 0x0823, 0x00 },
+	{ 0x3e20, 0x01 },
+	{ 0x3e37, 0x00 },
+	{ 0x3e3b, 0x01 },
+	{ 0x38a3, 0x01 },
+	{ 0x38a8, 0x00 },
+	{ 0x38a9, 0x00 },
+	{ 0x38aa, 0x00 },
+	{ 0x38ab, 0x00 },
+	{ 0x3234, 0x00 },
+	{ 0x3fc1, 0x00 },
+	{ 0x3235, 0x00 },
+	{ 0x3802, 0x00 },
+	{ 0x3143, 0x04 },
+	{ 0x360a, 0x00 },
+	{ 0x0b00, 0x00 },
+	{ 0x0106, 0x00 },
+	{ 0x0b05, 0x01 },
+	{ 0x0b06, 0x01 },
+	{ 0x3230, 0x00 },
+	{ 0x3602, 0x01 },
+	{ 0x3607, 0x01 },
+	{ 0x3c00, 0x00 },
+	{ 0x3c01, 0xba },
+	{ 0x3c02, 0xc8 },
+	{ 0x3c03, 0xaa },
+	{ 0x3c04, 0x91 },
+	{ 0x3c05, 0x54 },
+	{ 0x3c06, 0x26 },
+	{ 0x3c07, 0x20 },
+	{ 0x3c08, 0x51 },
+	{ 0x3d80, 0x00 },
+	{ 0x3f50, 0x00 },
+	{ 0x3f56, 0x00 },
+	{ 0x3f57, 0x30 },
+	{ 0x3f78, 0x00 },
+	{ 0x3f79, 0x34 },
+	{ 0x3f7c, 0x00 },
+	{ 0x3f7d, 0x00 },
+	{ 0x3fba, 0x00 },
+	{ 0x3fbb, 0x00 },
+	{ 0xa081, 0x04 },
+	{ 0xe014, 0x00 },
+	{ 0x0202, 0x05 },
+	{ 0x0203, 0x16 },
+	{ 0x0224, 0x01 },
+	{ 0x0225, 0xf4 },
+	{ 0x0204, 0x00 },
+	{ 0x0205, 0x00 },
+	{ 0x0216, 0x00 },
+	{ 0x0217, 0x00 },
+	{ 0x020e, 0x01 },
+	{ 0x020f, 0x00 },
+	{ 0x0210, 0x01 },
+	{ 0x0211, 0x00 },
+	{ 0x0212, 0x01 },
+	{ 0x0213, 0x00 },
+	{ 0x0214, 0x01 },
+	{ 0x0215, 0x00 },
+	{ 0x0218, 0x01 },
+	{ 0x0219, 0x00 },
+	{ 0x3614, 0x00 },
+	{ 0x3616, 0x0d },
+	{ 0x3617, 0x56 },
+	{ 0xb612, 0x20 },
+	{ 0xb613, 0x20 },
+	{ 0xb614, 0x20 },
+	{ 0xb615, 0x20 },
+	{ 0xb616, 0x0a },
+	{ 0xb617, 0x0a },
+	{ 0xb618, 0x20 },
+	{ 0xb619, 0x20 },
+	{ 0xb61a, 0x20 },
+	{ 0xb61b, 0x20 },
+	{ 0xb61c, 0x0a },
+	{ 0xb61d, 0x0a },
+	{ 0xb666, 0x30 },
+	{ 0xb667, 0x30 },
+	{ 0xb668, 0x30 },
+	{ 0xb669, 0x30 },
+	{ 0xb66a, 0x14 },
+	{ 0xb66b, 0x14 },
+	{ 0xb66c, 0x20 },
+	{ 0xb66d, 0x20 },
+	{ 0xb66e, 0x20 },
+	{ 0xb66f, 0x20 },
+	{ 0xb670, 0x10 },
+	{ 0xb671, 0x10 },
+	{ 0x3237, 0x00 },
+	{ 0x3900, 0x00 },
+	{ 0x3901, 0x00 },
+	{ 0x3902, 0x00 },
+	{ 0x3904, 0x00 },
+	{ 0x3905, 0x00 },
+	{ 0x3906, 0x00 },
+	{ 0x3907, 0x00 },
+	{ 0x3908, 0x00 },
+	{ 0x3909, 0x00 },
+	{ 0x3912, 0x00 },
+	{ 0x3930, 0x00 },
+	{ 0x3931, 0x00 },
+	{ 0x3933, 0x00 },
+	{ 0x3934, 0x00 },
+	{ 0x3935, 0x00 },
+	{ 0x3936, 0x00 },
+	{ 0x3937, 0x00 },
+	{ 0x30ac, 0x00 },
 };
 
 static const struct imx319_reg mode_1640x922_regs[] = {
-	{0x0112, 0x0a},
-	{0x0113, 0x0a},
-	{0x0114, 0x03},
-	{0x0342, 0x08},
-	{0x0343, 0x20},
-	{0x0340, 0x05},
-	{0x0341, 0x00},
-	{0x0344, 0x00},
-	{0x0345, 0x00},
-	{0x0346, 0x01},
-	{0x0347, 0x30},
-	{0x0348, 0x0c},
-	{0x0349, 0xcf},
-	{0x034a, 0x08},
-	{0x034b, 0x6f},
-	{0x0220, 0x00},
-	{0x0221, 0x11},
-	{0x0381, 0x01},
-	{0x0383, 0x01},
-	{0x0385, 0x01},
-	{0x0387, 0x01},
-	{0x0900, 0x01},
-	{0x0901, 0x22},
-	{0x0902, 0x0a},
-	{0x3140, 0x02},
-	{0x3141, 0x00},
-	{0x3f0d, 0x0a},
-	{0x3f14, 0x01},
-	{0x3f3c, 0x02},
-	{0x3f4d, 0x01},
-	{0x3f4c, 0x01},
-	{0x4254, 0x7f},
-	{0x0401, 0x00},
-	{0x0404, 0x00},
-	{0x0405, 0x10},
-	{0x0408, 0x00},
-	{0x0409, 0x00},
-	{0x040a, 0x00},
-	{0x040b, 0x02},
-	{0x040c, 0x06},
-	{0x040d, 0x68},
-	{0x040e, 0x03},
-	{0x040f, 0x9a},
-	{0x034c, 0x06},
-	{0x034d, 0x68},
-	{0x034e, 0x03},
-	{0x034f, 0x9a},
-	{0x3261, 0x00},
-	{0x3264, 0x00},
-	{0x3265, 0x10},
-	{0x0301, 0x06},
-	{0x0303, 0x04},
-	{0x0305, 0x03},
-	{0x0306, 0x01},
-	{0x0307, 0x2c},
-	{0x0309, 0x0a},
-	{0x030b, 0x02},
-	{0x030d, 0x04},
-	{0x030e, 0x01},
-	{0x030f, 0x2c},
-	{0x0310, 0x01},
-	{0x0820, 0x0b},
-	{0x0821, 0x40},
-	{0x0822, 0x00},
-	{0x0823, 0x00},
-	{0x3e20, 0x01},
-	{0x3e37, 0x00},
-	{0x3e3b, 0x01},
-	{0x38a3, 0x01},
-	{0x38a8, 0x00},
-	{0x38a9, 0x00},
-	{0x38aa, 0x00},
-	{0x38ab, 0x00},
-	{0x3234, 0x00},
-	{0x3fc1, 0x00},
-	{0x3235, 0x00},
-	{0x3802, 0x00},
-	{0x3143, 0x04},
-	{0x360a, 0x00},
-	{0x0b00, 0x00},
-	{0x0106, 0x00},
-	{0x0b05, 0x01},
-	{0x0b06, 0x01},
-	{0x3230, 0x00},
-	{0x3602, 0x01},
-	{0x3607, 0x01},
-	{0x3c00, 0x00},
-	{0x3c01, 0xba},
-	{0x3c02, 0xc8},
-	{0x3c03, 0xaa},
-	{0x3c04, 0x91},
-	{0x3c05, 0x54},
-	{0x3c06, 0x26},
-	{0x3c07, 0x20},
-	{0x3c08, 0x51},
-	{0x3d80, 0x00},
-	{0x3f50, 0x00},
-	{0x3f56, 0x00},
-	{0x3f57, 0x30},
-	{0x3f78, 0x00},
-	{0x3f79, 0x34},
-	{0x3f7c, 0x00},
-	{0x3f7d, 0x00},
-	{0x3fba, 0x00},
-	{0x3fbb, 0x00},
-	{0xa081, 0x04},
-	{0xe014, 0x00},
-	{0x0202, 0x04},
-	{0x0203, 0xee},
-	{0x0224, 0x01},
-	{0x0225, 0xf4},
-	{0x0204, 0x00},
-	{0x0205, 0x00},
-	{0x0216, 0x00},
-	{0x0217, 0x00},
-	{0x020e, 0x01},
-	{0x020f, 0x00},
-	{0x0210, 0x01},
-	{0x0211, 0x00},
-	{0x0212, 0x01},
-	{0x0213, 0x00},
-	{0x0214, 0x01},
-	{0x0215, 0x00},
-	{0x0218, 0x01},
-	{0x0219, 0x00},
-	{0x3614, 0x00},
-	{0x3616, 0x0d},
-	{0x3617, 0x56},
-	{0xb612, 0x20},
-	{0xb613, 0x20},
-	{0xb614, 0x20},
-	{0xb615, 0x20},
-	{0xb616, 0x0a},
-	{0xb617, 0x0a},
-	{0xb618, 0x20},
-	{0xb619, 0x20},
-	{0xb61a, 0x20},
-	{0xb61b, 0x20},
-	{0xb61c, 0x0a},
-	{0xb61d, 0x0a},
-	{0xb666, 0x30},
-	{0xb667, 0x30},
-	{0xb668, 0x30},
-	{0xb669, 0x30},
-	{0xb66a, 0x14},
-	{0xb66b, 0x14},
-	{0xb66c, 0x20},
-	{0xb66d, 0x20},
-	{0xb66e, 0x20},
-	{0xb66f, 0x20},
-	{0xb670, 0x10},
-	{0xb671, 0x10},
-	{0x3237, 0x00},
-	{0x3900, 0x00},
-	{0x3901, 0x00},
-	{0x3902, 0x00},
-	{0x3904, 0x00},
-	{0x3905, 0x00},
-	{0x3906, 0x00},
-	{0x3907, 0x00},
-	{0x3908, 0x00},
-	{0x3909, 0x00},
-	{0x3912, 0x00},
-	{0x3930, 0x00},
-	{0x3931, 0x00},
-	{0x3933, 0x00},
-	{0x3934, 0x00},
-	{0x3935, 0x00},
-	{0x3936, 0x00},
-	{0x3937, 0x00},
-	{0x30ac, 0x00},
+	{ 0x0112, 0x0a },
+	{ 0x0113, 0x0a },
+	{ 0x0114, 0x03 },
+	{ 0x0342, 0x08 },
+	{ 0x0343, 0x20 },
+	{ 0x0340, 0x05 },
+	{ 0x0341, 0x00 },
+	{ 0x0344, 0x00 },
+	{ 0x0345, 0x00 },
+	{ 0x0346, 0x01 },
+	{ 0x0347, 0x30 },
+	{ 0x0348, 0x0c },
+	{ 0x0349, 0xcf },
+	{ 0x034a, 0x08 },
+	{ 0x034b, 0x6f },
+	{ 0x0220, 0x00 },
+	{ 0x0221, 0x11 },
+	{ 0x0381, 0x01 },
+	{ 0x0383, 0x01 },
+	{ 0x0385, 0x01 },
+	{ 0x0387, 0x01 },
+	{ 0x0900, 0x01 },
+	{ 0x0901, 0x22 },
+	{ 0x0902, 0x0a },
+	{ 0x3140, 0x02 },
+	{ 0x3141, 0x00 },
+	{ 0x3f0d, 0x0a },
+	{ 0x3f14, 0x01 },
+	{ 0x3f3c, 0x02 },
+	{ 0x3f4d, 0x01 },
+	{ 0x3f4c, 0x01 },
+	{ 0x4254, 0x7f },
+	{ 0x0401, 0x00 },
+	{ 0x0404, 0x00 },
+	{ 0x0405, 0x10 },
+	{ 0x0408, 0x00 },
+	{ 0x0409, 0x00 },
+	{ 0x040a, 0x00 },
+	{ 0x040b, 0x02 },
+	{ 0x040c, 0x06 },
+	{ 0x040d, 0x68 },
+	{ 0x040e, 0x03 },
+	{ 0x040f, 0x9a },
+	{ 0x034c, 0x06 },
+	{ 0x034d, 0x68 },
+	{ 0x034e, 0x03 },
+	{ 0x034f, 0x9a },
+	{ 0x3261, 0x00 },
+	{ 0x3264, 0x00 },
+	{ 0x3265, 0x10 },
+	{ 0x0301, 0x06 },
+	{ 0x0303, 0x04 },
+	{ 0x0305, 0x03 },
+	{ 0x0306, 0x01 },
+	{ 0x0307, 0x2c },
+	{ 0x0309, 0x0a },
+	{ 0x030b, 0x02 },
+	{ 0x030d, 0x04 },
+	{ 0x030e, 0x01 },
+	{ 0x030f, 0x2c },
+	{ 0x0310, 0x01 },
+	{ 0x0820, 0x0b },
+	{ 0x0821, 0x40 },
+	{ 0x0822, 0x00 },
+	{ 0x0823, 0x00 },
+	{ 0x3e20, 0x01 },
+	{ 0x3e37, 0x00 },
+	{ 0x3e3b, 0x01 },
+	{ 0x38a3, 0x01 },
+	{ 0x38a8, 0x00 },
+	{ 0x38a9, 0x00 },
+	{ 0x38aa, 0x00 },
+	{ 0x38ab, 0x00 },
+	{ 0x3234, 0x00 },
+	{ 0x3fc1, 0x00 },
+	{ 0x3235, 0x00 },
+	{ 0x3802, 0x00 },
+	{ 0x3143, 0x04 },
+	{ 0x360a, 0x00 },
+	{ 0x0b00, 0x00 },
+	{ 0x0106, 0x00 },
+	{ 0x0b05, 0x01 },
+	{ 0x0b06, 0x01 },
+	{ 0x3230, 0x00 },
+	{ 0x3602, 0x01 },
+	{ 0x3607, 0x01 },
+	{ 0x3c00, 0x00 },
+	{ 0x3c01, 0xba },
+	{ 0x3c02, 0xc8 },
+	{ 0x3c03, 0xaa },
+	{ 0x3c04, 0x91 },
+	{ 0x3c05, 0x54 },
+	{ 0x3c06, 0x26 },
+	{ 0x3c07, 0x20 },
+	{ 0x3c08, 0x51 },
+	{ 0x3d80, 0x00 },
+	{ 0x3f50, 0x00 },
+	{ 0x3f56, 0x00 },
+	{ 0x3f57, 0x30 },
+	{ 0x3f78, 0x00 },
+	{ 0x3f79, 0x34 },
+	{ 0x3f7c, 0x00 },
+	{ 0x3f7d, 0x00 },
+	{ 0x3fba, 0x00 },
+	{ 0x3fbb, 0x00 },
+	{ 0xa081, 0x04 },
+	{ 0xe014, 0x00 },
+	{ 0x0202, 0x04 },
+	{ 0x0203, 0xee },
+	{ 0x0224, 0x01 },
+	{ 0x0225, 0xf4 },
+	{ 0x0204, 0x00 },
+	{ 0x0205, 0x00 },
+	{ 0x0216, 0x00 },
+	{ 0x0217, 0x00 },
+	{ 0x020e, 0x01 },
+	{ 0x020f, 0x00 },
+	{ 0x0210, 0x01 },
+	{ 0x0211, 0x00 },
+	{ 0x0212, 0x01 },
+	{ 0x0213, 0x00 },
+	{ 0x0214, 0x01 },
+	{ 0x0215, 0x00 },
+	{ 0x0218, 0x01 },
+	{ 0x0219, 0x00 },
+	{ 0x3614, 0x00 },
+	{ 0x3616, 0x0d },
+	{ 0x3617, 0x56 },
+	{ 0xb612, 0x20 },
+	{ 0xb613, 0x20 },
+	{ 0xb614, 0x20 },
+	{ 0xb615, 0x20 },
+	{ 0xb616, 0x0a },
+	{ 0xb617, 0x0a },
+	{ 0xb618, 0x20 },
+	{ 0xb619, 0x20 },
+	{ 0xb61a, 0x20 },
+	{ 0xb61b, 0x20 },
+	{ 0xb61c, 0x0a },
+	{ 0xb61d, 0x0a },
+	{ 0xb666, 0x30 },
+	{ 0xb667, 0x30 },
+	{ 0xb668, 0x30 },
+	{ 0xb669, 0x30 },
+	{ 0xb66a, 0x14 },
+	{ 0xb66b, 0x14 },
+	{ 0xb66c, 0x20 },
+	{ 0xb66d, 0x20 },
+	{ 0xb66e, 0x20 },
+	{ 0xb66f, 0x20 },
+	{ 0xb670, 0x10 },
+	{ 0xb671, 0x10 },
+	{ 0x3237, 0x00 },
+	{ 0x3900, 0x00 },
+	{ 0x3901, 0x00 },
+	{ 0x3902, 0x00 },
+	{ 0x3904, 0x00 },
+	{ 0x3905, 0x00 },
+	{ 0x3906, 0x00 },
+	{ 0x3907, 0x00 },
+	{ 0x3908, 0x00 },
+	{ 0x3909, 0x00 },
+	{ 0x3912, 0x00 },
+	{ 0x3930, 0x00 },
+	{ 0x3931, 0x00 },
+	{ 0x3933, 0x00 },
+	{ 0x3934, 0x00 },
+	{ 0x3935, 0x00 },
+	{ 0x3936, 0x00 },
+	{ 0x3937, 0x00 },
+	{ 0x30ac, 0x00 },
 };
 
 static const struct imx319_reg mode_1296x736_regs[] = {
-	{0x0112, 0x0a},
-	{0x0113, 0x0a},
-	{0x0114, 0x03},
-	{0x0342, 0x08},
-	{0x0343, 0x20},
-	{0x0340, 0x05},
-	{0x0341, 0x00},
-	{0x0344, 0x00},
-	{0x0345, 0x00},
-	{0x0346, 0x01},
-	{0x0347, 0xf0},
-	{0x0348, 0x0c},
-	{0x0349, 0xcf},
-	{0x034a, 0x07},
-	{0x034b, 0xaf},
-	{0x0220, 0x00},
-	{0x0221, 0x11},
-	{0x0381, 0x01},
-	{0x0383, 0x01},
-	{0x0385, 0x01},
-	{0x0387, 0x01},
-	{0x0900, 0x01},
-	{0x0901, 0x22},
-	{0x0902, 0x0a},
-	{0x3140, 0x02},
-	{0x3141, 0x00},
-	{0x3f0d, 0x0a},
-	{0x3f14, 0x01},
-	{0x3f3c, 0x02},
-	{0x3f4d, 0x01},
-	{0x3f4c, 0x01},
-	{0x4254, 0x7f},
-	{0x0401, 0x00},
-	{0x0404, 0x00},
-	{0x0405, 0x10},
-	{0x0408, 0x00},
-	{0x0409, 0xac},
-	{0x040a, 0x00},
-	{0x040b, 0x00},
-	{0x040c, 0x05},
-	{0x040d, 0x10},
-	{0x040e, 0x02},
-	{0x040f, 0xe0},
-	{0x034c, 0x05},
-	{0x034d, 0x10},
-	{0x034e, 0x02},
-	{0x034f, 0xe0},
-	{0x3261, 0x00},
-	{0x3264, 0x00},
-	{0x3265, 0x10},
-	{0x0301, 0x06},
-	{0x0303, 0x04},
-	{0x0305, 0x03},
-	{0x0306, 0x01},
-	{0x0307, 0x2c},
-	{0x0309, 0x0a},
-	{0x030b, 0x02},
-	{0x030d, 0x04},
-	{0x030e, 0x01},
-	{0x030f, 0x2c},
-	{0x0310, 0x01},
-	{0x0820, 0x0b},
-	{0x0821, 0x40},
-	{0x0822, 0x00},
-	{0x0823, 0x00},
-	{0x3e20, 0x01},
-	{0x3e37, 0x00},
-	{0x3e3b, 0x01},
-	{0x38a3, 0x01},
-	{0x38a8, 0x00},
-	{0x38a9, 0x00},
-	{0x38aa, 0x00},
-	{0x38ab, 0x00},
-	{0x3234, 0x00},
-	{0x3fc1, 0x00},
-	{0x3235, 0x00},
-	{0x3802, 0x00},
-	{0x3143, 0x04},
-	{0x360a, 0x00},
-	{0x0b00, 0x00},
-	{0x0106, 0x00},
-	{0x0b05, 0x01},
-	{0x0b06, 0x01},
-	{0x3230, 0x00},
-	{0x3602, 0x01},
-	{0x3607, 0x01},
-	{0x3c00, 0x00},
-	{0x3c01, 0xba},
-	{0x3c02, 0xc8},
-	{0x3c03, 0xaa},
-	{0x3c04, 0x91},
-	{0x3c05, 0x54},
-	{0x3c06, 0x26},
-	{0x3c07, 0x20},
-	{0x3c08, 0x51},
-	{0x3d80, 0x00},
-	{0x3f50, 0x00},
-	{0x3f56, 0x00},
-	{0x3f57, 0x30},
-	{0x3f78, 0x00},
-	{0x3f79, 0x34},
-	{0x3f7c, 0x00},
-	{0x3f7d, 0x00},
-	{0x3fba, 0x00},
-	{0x3fbb, 0x00},
-	{0xa081, 0x04},
-	{0xe014, 0x00},
-	{0x0202, 0x04},
-	{0x0203, 0xee},
-	{0x0224, 0x01},
-	{0x0225, 0xf4},
-	{0x0204, 0x00},
-	{0x0205, 0x00},
-	{0x0216, 0x00},
-	{0x0217, 0x00},
-	{0x020e, 0x01},
-	{0x020f, 0x00},
-	{0x0210, 0x01},
-	{0x0211, 0x00},
-	{0x0212, 0x01},
-	{0x0213, 0x00},
-	{0x0214, 0x01},
-	{0x0215, 0x00},
-	{0x0218, 0x01},
-	{0x0219, 0x00},
-	{0x3614, 0x00},
-	{0x3616, 0x0d},
-	{0x3617, 0x56},
-	{0xb612, 0x20},
-	{0xb613, 0x20},
-	{0xb614, 0x20},
-	{0xb615, 0x20},
-	{0xb616, 0x0a},
-	{0xb617, 0x0a},
-	{0xb618, 0x20},
-	{0xb619, 0x20},
-	{0xb61a, 0x20},
-	{0xb61b, 0x20},
-	{0xb61c, 0x0a},
-	{0xb61d, 0x0a},
-	{0xb666, 0x30},
-	{0xb667, 0x30},
-	{0xb668, 0x30},
-	{0xb669, 0x30},
-	{0xb66a, 0x14},
-	{0xb66b, 0x14},
-	{0xb66c, 0x20},
-	{0xb66d, 0x20},
-	{0xb66e, 0x20},
-	{0xb66f, 0x20},
-	{0xb670, 0x10},
-	{0xb671, 0x10},
-	{0x3237, 0x00},
-	{0x3900, 0x00},
-	{0x3901, 0x00},
-	{0x3902, 0x00},
-	{0x3904, 0x00},
-	{0x3905, 0x00},
-	{0x3906, 0x00},
-	{0x3907, 0x00},
-	{0x3908, 0x00},
-	{0x3909, 0x00},
-	{0x3912, 0x00},
-	{0x3930, 0x00},
-	{0x3931, 0x00},
-	{0x3933, 0x00},
-	{0x3934, 0x00},
-	{0x3935, 0x00},
-	{0x3936, 0x00},
-	{0x3937, 0x00},
-	{0x30ac, 0x00},
+	{ 0x0112, 0x0a },
+	{ 0x0113, 0x0a },
+	{ 0x0114, 0x03 },
+	{ 0x0342, 0x08 },
+	{ 0x0343, 0x20 },
+	{ 0x0340, 0x05 },
+	{ 0x0341, 0x00 },
+	{ 0x0344, 0x00 },
+	{ 0x0345, 0x00 },
+	{ 0x0346, 0x01 },
+	{ 0x0347, 0xf0 },
+	{ 0x0348, 0x0c },
+	{ 0x0349, 0xcf },
+	{ 0x034a, 0x07 },
+	{ 0x034b, 0xaf },
+	{ 0x0220, 0x00 },
+	{ 0x0221, 0x11 },
+	{ 0x0381, 0x01 },
+	{ 0x0383, 0x01 },
+	{ 0x0385, 0x01 },
+	{ 0x0387, 0x01 },
+	{ 0x0900, 0x01 },
+	{ 0x0901, 0x22 },
+	{ 0x0902, 0x0a },
+	{ 0x3140, 0x02 },
+	{ 0x3141, 0x00 },
+	{ 0x3f0d, 0x0a },
+	{ 0x3f14, 0x01 },
+	{ 0x3f3c, 0x02 },
+	{ 0x3f4d, 0x01 },
+	{ 0x3f4c, 0x01 },
+	{ 0x4254, 0x7f },
+	{ 0x0401, 0x00 },
+	{ 0x0404, 0x00 },
+	{ 0x0405, 0x10 },
+	{ 0x0408, 0x00 },
+	{ 0x0409, 0xac },
+	{ 0x040a, 0x00 },
+	{ 0x040b, 0x00 },
+	{ 0x040c, 0x05 },
+	{ 0x040d, 0x10 },
+	{ 0x040e, 0x02 },
+	{ 0x040f, 0xe0 },
+	{ 0x034c, 0x05 },
+	{ 0x034d, 0x10 },
+	{ 0x034e, 0x02 },
+	{ 0x034f, 0xe0 },
+	{ 0x3261, 0x00 },
+	{ 0x3264, 0x00 },
+	{ 0x3265, 0x10 },
+	{ 0x0301, 0x06 },
+	{ 0x0303, 0x04 },
+	{ 0x0305, 0x03 },
+	{ 0x0306, 0x01 },
+	{ 0x0307, 0x2c },
+	{ 0x0309, 0x0a },
+	{ 0x030b, 0x02 },
+	{ 0x030d, 0x04 },
+	{ 0x030e, 0x01 },
+	{ 0x030f, 0x2c },
+	{ 0x0310, 0x01 },
+	{ 0x0820, 0x0b },
+	{ 0x0821, 0x40 },
+	{ 0x0822, 0x00 },
+	{ 0x0823, 0x00 },
+	{ 0x3e20, 0x01 },
+	{ 0x3e37, 0x00 },
+	{ 0x3e3b, 0x01 },
+	{ 0x38a3, 0x01 },
+	{ 0x38a8, 0x00 },
+	{ 0x38a9, 0x00 },
+	{ 0x38aa, 0x00 },
+	{ 0x38ab, 0x00 },
+	{ 0x3234, 0x00 },
+	{ 0x3fc1, 0x00 },
+	{ 0x3235, 0x00 },
+	{ 0x3802, 0x00 },
+	{ 0x3143, 0x04 },
+	{ 0x360a, 0x00 },
+	{ 0x0b00, 0x00 },
+	{ 0x0106, 0x00 },
+	{ 0x0b05, 0x01 },
+	{ 0x0b06, 0x01 },
+	{ 0x3230, 0x00 },
+	{ 0x3602, 0x01 },
+	{ 0x3607, 0x01 },
+	{ 0x3c00, 0x00 },
+	{ 0x3c01, 0xba },
+	{ 0x3c02, 0xc8 },
+	{ 0x3c03, 0xaa },
+	{ 0x3c04, 0x91 },
+	{ 0x3c05, 0x54 },
+	{ 0x3c06, 0x26 },
+	{ 0x3c07, 0x20 },
+	{ 0x3c08, 0x51 },
+	{ 0x3d80, 0x00 },
+	{ 0x3f50, 0x00 },
+	{ 0x3f56, 0x00 },
+	{ 0x3f57, 0x30 },
+	{ 0x3f78, 0x00 },
+	{ 0x3f79, 0x34 },
+	{ 0x3f7c, 0x00 },
+	{ 0x3f7d, 0x00 },
+	{ 0x3fba, 0x00 },
+	{ 0x3fbb, 0x00 },
+	{ 0xa081, 0x04 },
+	{ 0xe014, 0x00 },
+	{ 0x0202, 0x04 },
+	{ 0x0203, 0xee },
+	{ 0x0224, 0x01 },
+	{ 0x0225, 0xf4 },
+	{ 0x0204, 0x00 },
+	{ 0x0205, 0x00 },
+	{ 0x0216, 0x00 },
+	{ 0x0217, 0x00 },
+	{ 0x020e, 0x01 },
+	{ 0x020f, 0x00 },
+	{ 0x0210, 0x01 },
+	{ 0x0211, 0x00 },
+	{ 0x0212, 0x01 },
+	{ 0x0213, 0x00 },
+	{ 0x0214, 0x01 },
+	{ 0x0215, 0x00 },
+	{ 0x0218, 0x01 },
+	{ 0x0219, 0x00 },
+	{ 0x3614, 0x00 },
+	{ 0x3616, 0x0d },
+	{ 0x3617, 0x56 },
+	{ 0xb612, 0x20 },
+	{ 0xb613, 0x20 },
+	{ 0xb614, 0x20 },
+	{ 0xb615, 0x20 },
+	{ 0xb616, 0x0a },
+	{ 0xb617, 0x0a },
+	{ 0xb618, 0x20 },
+	{ 0xb619, 0x20 },
+	{ 0xb61a, 0x20 },
+	{ 0xb61b, 0x20 },
+	{ 0xb61c, 0x0a },
+	{ 0xb61d, 0x0a },
+	{ 0xb666, 0x30 },
+	{ 0xb667, 0x30 },
+	{ 0xb668, 0x30 },
+	{ 0xb669, 0x30 },
+	{ 0xb66a, 0x14 },
+	{ 0xb66b, 0x14 },
+	{ 0xb66c, 0x20 },
+	{ 0xb66d, 0x20 },
+	{ 0xb66e, 0x20 },
+	{ 0xb66f, 0x20 },
+	{ 0xb670, 0x10 },
+	{ 0xb671, 0x10 },
+	{ 0x3237, 0x00 },
+	{ 0x3900, 0x00 },
+	{ 0x3901, 0x00 },
+	{ 0x3902, 0x00 },
+	{ 0x3904, 0x00 },
+	{ 0x3905, 0x00 },
+	{ 0x3906, 0x00 },
+	{ 0x3907, 0x00 },
+	{ 0x3908, 0x00 },
+	{ 0x3909, 0x00 },
+	{ 0x3912, 0x00 },
+	{ 0x3930, 0x00 },
+	{ 0x3931, 0x00 },
+	{ 0x3933, 0x00 },
+	{ 0x3934, 0x00 },
+	{ 0x3935, 0x00 },
+	{ 0x3936, 0x00 },
+	{ 0x3937, 0x00 },
+	{ 0x30ac, 0x00 },
 };
 
 static const struct imx319_reg mode_1280x720_regs[] = {
-	{0x0112, 0x0a},
-	{0x0113, 0x0a},
-	{0x0114, 0x03},
-	{0x0342, 0x08},
-	{0x0343, 0x20},
-	{0x0340, 0x05},
-	{0x0341, 0x00},
-	{0x0344, 0x00},
-	{0x0345, 0x00},
-	{0x0346, 0x02},
-	{0x0347, 0x00},
-	{0x0348, 0x0c},
-	{0x0349, 0xcf},
-	{0x034a, 0x07},
-	{0x034b, 0x9f},
-	{0x0220, 0x00},
-	{0x0221, 0x11},
-	{0x0381, 0x01},
-	{0x0383, 0x01},
-	{0x0385, 0x01},
-	{0x0387, 0x01},
-	{0x0900, 0x01},
-	{0x0901, 0x22},
-	{0x0902, 0x0a},
-	{0x3140, 0x02},
-	{0x3141, 0x00},
-	{0x3f0d, 0x0a},
-	{0x3f14, 0x01},
-	{0x3f3c, 0x02},
-	{0x3f4d, 0x01},
-	{0x3f4c, 0x01},
-	{0x4254, 0x7f},
-	{0x0401, 0x00},
-	{0x0404, 0x00},
-	{0x0405, 0x10},
-	{0x0408, 0x00},
-	{0x0409, 0xb4},
-	{0x040a, 0x00},
-	{0x040b, 0x00},
-	{0x040c, 0x05},
-	{0x040d, 0x00},
-	{0x040e, 0x02},
-	{0x040f, 0xd0},
-	{0x034c, 0x05},
-	{0x034d, 0x00},
-	{0x034e, 0x02},
-	{0x034f, 0xd0},
-	{0x3261, 0x00},
-	{0x3264, 0x00},
-	{0x3265, 0x10},
-	{0x0301, 0x06},
-	{0x0303, 0x04},
-	{0x0305, 0x03},
-	{0x0306, 0x01},
-	{0x0307, 0x2c},
-	{0x0309, 0x0a},
-	{0x030b, 0x02},
-	{0x030d, 0x04},
-	{0x030e, 0x01},
-	{0x030f, 0x2c},
-	{0x0310, 0x01},
-	{0x0820, 0x0b},
-	{0x0821, 0x40},
-	{0x0822, 0x00},
-	{0x0823, 0x00},
-	{0x3e20, 0x01},
-	{0x3e37, 0x00},
-	{0x3e3b, 0x01},
-	{0x38a3, 0x01},
-	{0x38a8, 0x00},
-	{0x38a9, 0x00},
-	{0x38aa, 0x00},
-	{0x38ab, 0x00},
-	{0x3234, 0x00},
-	{0x3fc1, 0x00},
-	{0x3235, 0x00},
-	{0x3802, 0x00},
-	{0x3143, 0x04},
-	{0x360a, 0x00},
-	{0x0b00, 0x00},
-	{0x0106, 0x00},
-	{0x0b05, 0x01},
-	{0x0b06, 0x01},
-	{0x3230, 0x00},
-	{0x3602, 0x01},
-	{0x3607, 0x01},
-	{0x3c00, 0x00},
-	{0x3c01, 0xba},
-	{0x3c02, 0xc8},
-	{0x3c03, 0xaa},
-	{0x3c04, 0x91},
-	{0x3c05, 0x54},
-	{0x3c06, 0x26},
-	{0x3c07, 0x20},
-	{0x3c08, 0x51},
-	{0x3d80, 0x00},
-	{0x3f50, 0x00},
-	{0x3f56, 0x00},
-	{0x3f57, 0x30},
-	{0x3f78, 0x00},
-	{0x3f79, 0x34},
-	{0x3f7c, 0x00},
-	{0x3f7d, 0x00},
-	{0x3fba, 0x00},
-	{0x3fbb, 0x00},
-	{0xa081, 0x04},
-	{0xe014, 0x00},
-	{0x0202, 0x04},
-	{0x0203, 0xee},
-	{0x0224, 0x01},
-	{0x0225, 0xf4},
-	{0x0204, 0x00},
-	{0x0205, 0x00},
-	{0x0216, 0x00},
-	{0x0217, 0x00},
-	{0x020e, 0x01},
-	{0x020f, 0x00},
-	{0x0210, 0x01},
-	{0x0211, 0x00},
-	{0x0212, 0x01},
-	{0x0213, 0x00},
-	{0x0214, 0x01},
-	{0x0215, 0x00},
-	{0x0218, 0x01},
-	{0x0219, 0x00},
-	{0x3614, 0x00},
-	{0x3616, 0x0d},
-	{0x3617, 0x56},
-	{0xb612, 0x20},
-	{0xb613, 0x20},
-	{0xb614, 0x20},
-	{0xb615, 0x20},
-	{0xb616, 0x0a},
-	{0xb617, 0x0a},
-	{0xb618, 0x20},
-	{0xb619, 0x20},
-	{0xb61a, 0x20},
-	{0xb61b, 0x20},
-	{0xb61c, 0x0a},
-	{0xb61d, 0x0a},
-	{0xb666, 0x30},
-	{0xb667, 0x30},
-	{0xb668, 0x30},
-	{0xb669, 0x30},
-	{0xb66a, 0x14},
-	{0xb66b, 0x14},
-	{0xb66c, 0x20},
-	{0xb66d, 0x20},
-	{0xb66e, 0x20},
-	{0xb66f, 0x20},
-	{0xb670, 0x10},
-	{0xb671, 0x10},
-	{0x3237, 0x00},
-	{0x3900, 0x00},
-	{0x3901, 0x00},
-	{0x3902, 0x00},
-	{0x3904, 0x00},
-	{0x3905, 0x00},
-	{0x3906, 0x00},
-	{0x3907, 0x00},
-	{0x3908, 0x00},
-	{0x3909, 0x00},
-	{0x3912, 0x00},
-	{0x3930, 0x00},
-	{0x3931, 0x00},
-	{0x3933, 0x00},
-	{0x3934, 0x00},
-	{0x3935, 0x00},
-	{0x3936, 0x00},
-	{0x3937, 0x00},
-	{0x30ac, 0x00},
+	{ 0x0112, 0x0a },
+	{ 0x0113, 0x0a },
+	{ 0x0114, 0x03 },
+	{ 0x0342, 0x08 },
+	{ 0x0343, 0x20 },
+	{ 0x0340, 0x05 },
+	{ 0x0341, 0x00 },
+	{ 0x0344, 0x00 },
+	{ 0x0345, 0x00 },
+	{ 0x0346, 0x02 },
+	{ 0x0347, 0x00 },
+	{ 0x0348, 0x0c },
+	{ 0x0349, 0xcf },
+	{ 0x034a, 0x07 },
+	{ 0x034b, 0x9f },
+	{ 0x0220, 0x00 },
+	{ 0x0221, 0x11 },
+	{ 0x0381, 0x01 },
+	{ 0x0383, 0x01 },
+	{ 0x0385, 0x01 },
+	{ 0x0387, 0x01 },
+	{ 0x0900, 0x01 },
+	{ 0x0901, 0x22 },
+	{ 0x0902, 0x0a },
+	{ 0x3140, 0x02 },
+	{ 0x3141, 0x00 },
+	{ 0x3f0d, 0x0a },
+	{ 0x3f14, 0x01 },
+	{ 0x3f3c, 0x02 },
+	{ 0x3f4d, 0x01 },
+	{ 0x3f4c, 0x01 },
+	{ 0x4254, 0x7f },
+	{ 0x0401, 0x00 },
+	{ 0x0404, 0x00 },
+	{ 0x0405, 0x10 },
+	{ 0x0408, 0x00 },
+	{ 0x0409, 0xb4 },
+	{ 0x040a, 0x00 },
+	{ 0x040b, 0x00 },
+	{ 0x040c, 0x05 },
+	{ 0x040d, 0x00 },
+	{ 0x040e, 0x02 },
+	{ 0x040f, 0xd0 },
+	{ 0x034c, 0x05 },
+	{ 0x034d, 0x00 },
+	{ 0x034e, 0x02 },
+	{ 0x034f, 0xd0 },
+	{ 0x3261, 0x00 },
+	{ 0x3264, 0x00 },
+	{ 0x3265, 0x10 },
+	{ 0x0301, 0x06 },
+	{ 0x0303, 0x04 },
+	{ 0x0305, 0x03 },
+	{ 0x0306, 0x01 },
+	{ 0x0307, 0x2c },
+	{ 0x0309, 0x0a },
+	{ 0x030b, 0x02 },
+	{ 0x030d, 0x04 },
+	{ 0x030e, 0x01 },
+	{ 0x030f, 0x2c },
+	{ 0x0310, 0x01 },
+	{ 0x0820, 0x0b },
+	{ 0x0821, 0x40 },
+	{ 0x0822, 0x00 },
+	{ 0x0823, 0x00 },
+	{ 0x3e20, 0x01 },
+	{ 0x3e37, 0x00 },
+	{ 0x3e3b, 0x01 },
+	{ 0x38a3, 0x01 },
+	{ 0x38a8, 0x00 },
+	{ 0x38a9, 0x00 },
+	{ 0x38aa, 0x00 },
+	{ 0x38ab, 0x00 },
+	{ 0x3234, 0x00 },
+	{ 0x3fc1, 0x00 },
+	{ 0x3235, 0x00 },
+	{ 0x3802, 0x00 },
+	{ 0x3143, 0x04 },
+	{ 0x360a, 0x00 },
+	{ 0x0b00, 0x00 },
+	{ 0x0106, 0x00 },
+	{ 0x0b05, 0x01 },
+	{ 0x0b06, 0x01 },
+	{ 0x3230, 0x00 },
+	{ 0x3602, 0x01 },
+	{ 0x3607, 0x01 },
+	{ 0x3c00, 0x00 },
+	{ 0x3c01, 0xba },
+	{ 0x3c02, 0xc8 },
+	{ 0x3c03, 0xaa },
+	{ 0x3c04, 0x91 },
+	{ 0x3c05, 0x54 },
+	{ 0x3c06, 0x26 },
+	{ 0x3c07, 0x20 },
+	{ 0x3c08, 0x51 },
+	{ 0x3d80, 0x00 },
+	{ 0x3f50, 0x00 },
+	{ 0x3f56, 0x00 },
+	{ 0x3f57, 0x30 },
+	{ 0x3f78, 0x00 },
+	{ 0x3f79, 0x34 },
+	{ 0x3f7c, 0x00 },
+	{ 0x3f7d, 0x00 },
+	{ 0x3fba, 0x00 },
+	{ 0x3fbb, 0x00 },
+	{ 0xa081, 0x04 },
+	{ 0xe014, 0x00 },
+	{ 0x0202, 0x04 },
+	{ 0x0203, 0xee },
+	{ 0x0224, 0x01 },
+	{ 0x0225, 0xf4 },
+	{ 0x0204, 0x00 },
+	{ 0x0205, 0x00 },
+	{ 0x0216, 0x00 },
+	{ 0x0217, 0x00 },
+	{ 0x020e, 0x01 },
+	{ 0x020f, 0x00 },
+	{ 0x0210, 0x01 },
+	{ 0x0211, 0x00 },
+	{ 0x0212, 0x01 },
+	{ 0x0213, 0x00 },
+	{ 0x0214, 0x01 },
+	{ 0x0215, 0x00 },
+	{ 0x0218, 0x01 },
+	{ 0x0219, 0x00 },
+	{ 0x3614, 0x00 },
+	{ 0x3616, 0x0d },
+	{ 0x3617, 0x56 },
+	{ 0xb612, 0x20 },
+	{ 0xb613, 0x20 },
+	{ 0xb614, 0x20 },
+	{ 0xb615, 0x20 },
+	{ 0xb616, 0x0a },
+	{ 0xb617, 0x0a },
+	{ 0xb618, 0x20 },
+	{ 0xb619, 0x20 },
+	{ 0xb61a, 0x20 },
+	{ 0xb61b, 0x20 },
+	{ 0xb61c, 0x0a },
+	{ 0xb61d, 0x0a },
+	{ 0xb666, 0x30 },
+	{ 0xb667, 0x30 },
+	{ 0xb668, 0x30 },
+	{ 0xb669, 0x30 },
+	{ 0xb66a, 0x14 },
+	{ 0xb66b, 0x14 },
+	{ 0xb66c, 0x20 },
+	{ 0xb66d, 0x20 },
+	{ 0xb66e, 0x20 },
+	{ 0xb66f, 0x20 },
+	{ 0xb670, 0x10 },
+	{ 0xb671, 0x10 },
+	{ 0x3237, 0x00 },
+	{ 0x3900, 0x00 },
+	{ 0x3901, 0x00 },
+	{ 0x3902, 0x00 },
+	{ 0x3904, 0x00 },
+	{ 0x3905, 0x00 },
+	{ 0x3906, 0x00 },
+	{ 0x3907, 0x00 },
+	{ 0x3908, 0x00 },
+	{ 0x3909, 0x00 },
+	{ 0x3912, 0x00 },
+	{ 0x3930, 0x00 },
+	{ 0x3931, 0x00 },
+	{ 0x3933, 0x00 },
+	{ 0x3934, 0x00 },
+	{ 0x3935, 0x00 },
+	{ 0x3936, 0x00 },
+	{ 0x3937, 0x00 },
+	{ 0x30ac, 0x00 },
 };
 
 static const char * const imx319_test_pattern_menu[] = {
@@ -2495,10 +2495,9 @@ static int imx319_remove(struct i2c_client *client)
 }
 
 static const struct i2c_device_id imx319_id_table[] = {
-	{"imx319", 0},
-	{},
+	{ "imx319", 0 },
+	{ /* sentinel */ },
 };
-
 MODULE_DEVICE_TABLE(i2c, imx319_id_table);
 
 static const struct dev_pm_ops imx319_pm_ops = {
@@ -2507,10 +2506,9 @@ static int imx319_remove(struct i2c_client *client)
 
 #ifdef CONFIG_ACPI
 static const struct acpi_device_id imx319_acpi_ids[] = {
-	{"IMX319"},
+	{ "IMX319" },
 	{ /* sentinel */ }
 };
-
 MODULE_DEVICE_TABLE(acpi, imx319_acpi_ids);
 #endif
 
@@ -2525,7 +2523,6 @@ static int imx319_remove(struct i2c_client *client)
 	.remove = imx319_remove,
 	.id_table = imx319_id_table,
 };
-
 module_i2c_driver(imx319_i2c_driver);
 
 MODULE_AUTHOR("Qiu, Tianshu <tian.shu.qiu@intel.com>");
-- 
1.9.1

